[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/macmanes-lab/DoveParentsRNAseq/master?urlpath=rstudio)

# Characterizing the neurogenomics of parental care in the rock dove

## Overview

This repository contains the data and analysis for a collaboration between Drs. Rebecca Calisi and Matt MacManes that focuses on one characterizing the neurogenomocs of parental care in the rock dove.


## Organization

These repositories is broken down into the following sub-repositories, each with their own unique purpose and structure.

- analysis: where the .Rmd script and the .md outputs live. The prefix corresponds to the order of operation. 
- figures: where figure generated by the scripts live. The prefix correspond to the script prefix that created the files
- metadata: contains files that describe the sample variables and a rosseta stone for transcripts to gene ids. 
- results: where data generated by the scripts live. The prefix correspond to the script prefix that created the files

There are two hidden directories, `kallisto_mappings` and `mapping`, which contain the results of the kallisto and salmon algorithms that transform read counts into gene counts.  

## Figures and Tables


## Table 1: Candidate gene DEGs

<table>
<thead>
<tr>
<th style="text-align:left;">
sex
</th>
<th style="text-align:left;">
tissue
</th>
<th style="text-align:left;">
bldg\_lay
</th>
<th style="text-align:left;">
lay\_inc.d3
</th>
<th style="text-align:left;">
inc.d3\_inc.d9
</th>
<th style="text-align:left;">
inc.d9\_inc.d17
</th>
<th style="text-align:left;">
inc.d17\_hatch
</th>
<th style="text-align:left;">
hatch\_n5
</th>
<th style="text-align:left;">
n5\_n9
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
female
</td>
<td style="text-align:left;">
hypothalamus
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
BDNF BRCA1 CISH CYP19A1 DRD1 EGR1 GRIA1 POMC
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
male
</td>
<td style="text-align:left;">
hypothalamus
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
female
</td>
<td style="text-align:left;">
pituitary
</td>
<td style="text-align:left;">
DIO3 ESR1 GNRHR
</td>
<td style="text-align:left;">
ESR1 GNRHR PTEN ZBTB16
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
BRCA1 CDK1 DIO2 KCNJ5 LBH PRL
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AVPR2 BRCA1 CDK1 GRIA1 LBH PRL
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
male
</td>
<td style="text-align:left;">
pituitary
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
BRCA1 CDK1 CISH GRM2 LBH PRL VIP
</td>
<td style="text-align:left;">
GRM2
</td>
<td style="text-align:left;">
CEBPD ZBTB16
</td>
<td style="text-align:left;">
BRCA1 CDK1 CEBPD
</td>
</tr>
<tr>
<td style="text-align:left;">
female
</td>
<td style="text-align:left;">
gonad
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
AGRP AVPR1A DIO2 EGR1 FOS PRLR PTGER3
</td>
<td style="text-align:left;">
AVPR1A
</td>
<td style="text-align:left;">
SOX9
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
male
</td>
<td style="text-align:left;">
gonad
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
SOX9
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
</tbody>
</table>


## Figure 1: Overall pattern of gene expression

![](./figures/fig1-1.png)
## Figure 2: Sequential differential gene expression in the HPG

![](./figures/fig2-1.png)

## Figure 3: Prolactin gene expression in the pituitary explains most of the variation in the whole data set and is correlated with circulating prolactin. In the pituitary, more chagnes gene expression are due to changes in internal _PRL_ expression (high versus lo) than the external enviornment (eggs versus chicks)

![](./figures/fig3-1.png)

## Figure 5: manipualations, work in progess...

![](./figures/fig5-1.png)



## Related documentation.

Another website <http://www.dovelovegenomics.org/>

Another GitHub repository <https://github.com/macmanes-lab/RockDove> 

A talk from 2020 <https://speakerdeck.com/raynamharris/peaks-and-valleys-of-prolactin-driven-gene-expression-during-parental-care>

An poster from 2019
![poster](./figures/sbnposter/SBN.png)

See also <https://macmanes-lab.github.io/DoveParentsRNAseq/> for a more user-friendly way to read the information on the poster. The bulk of the analysis for the SBN poster were created using the script [analysis/04_DESeq2_sexes.Rmd](https://github.com/macmanes-lab/DoveParentsRNAseq/blob/master/analysis/04_DESeq2_sexes.Rmd), which really needs a different name. 