---
title: "00_datawrangling.Rmd"
output: md_document
---

```{r setup}
# https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html
library(tidyverse)
```

# Data Wrangling

## countData and geneinfo

```{r countData}
# import count data, set rows at entreziz
kallistodata <- read.table("../results/kallistocounts.txt", 
                        sep = ",", row.names = NULL)
head(kallistodata)

## save gene information
geneinfo <- kallistodata %>%
  select(Name, geneid, entrezid)
head(geneinfo)


# check for genes that have multple transcripts expressed
isoforms <- kallistodata %>%
  group_by(Name) %>%
  summarize(n = n()) %>%
  arrange(desc(n), Name) %>%
  filter(n > 1)
head(isoforms)

# for example, this gene has 
geneinfo %>% filter(Name == "GRIN1")

# aggregate transcript counts to gene counts
countData <- kallistodata %>% 
  select(-row.names, -geneid, -entrezid) %>% 
  rename(gene = Name) %>%
  pivot_longer(-gene, names_to = "samples", values_to = "counts") %>%  
  pivot_wider(
    names_from = samples, 
    values_from = counts,
    values_fn = list(counts = sum))  %>% 
  arrange(gene) %>% 
  filter(gene != "")
head(countData)

countData <- as.data.frame(countData)
row.names(countData) <- countData$gene ## make gene the row name
countData[1] <- NULL ## make gene the row name
countData <- round(countData) #round all value to nearest 1s place
head(countData)

# print tolal num of genes and samples
dim(countData)
```

## wrangle colData 

```{r colData}
colData <- read.table(file.path( "../metadata/kallistosamples.txt"), header = F, stringsAsFactors = F)

# use strsplit to cut the filename into meaningful columns
colData$bird <- sapply(strsplit(as.character(colData$V1),'\\_'), "[", 1)
colData$sex <- sapply(strsplit(as.character(colData$V1),'\\_'), "[", 2)
colData$tissue <- sapply(strsplit(as.character(colData$V1),'\\_'), "[", 3)

# to create columns with variable names in the filename 
colData$temp <- sapply(strsplit(as.character(colData$V1),'\\_'), "[", 4)
colData$treatmentTemp <- sapply(strsplit(as.character(colData$temp),'\\.'), "[", 1)
colData$NYNO <- sapply(strsplit(as.character(colData$temp),'\\.'), "[", 2)

# relabel incorrectly named variables 
colData$treatment <- ifelse(grepl("extend-hatch", colData$treatmentTemp), "extend",
                            ifelse(grepl("inc-prolong", colData$treatmentTemp), "prolong",
                                   ifelse(grepl("m.hatch", colData$treatmentTemp), "m.n2",
                                   colData$treatmentTemp)))

# drop no-longer-needed cols
colData <- colData %>% dplyr::select(-c(temp, treatmentTemp, NYNO))

# replace dashes with periods and make rowname, (to make deseq happy)
colData$bird <- gsub("-", ".", colData$bird)
colData$treatment <- gsub("-", ".", colData$treatment)
colData$V1 <- gsub("-", ".", colData$V1)
row.names(colData) <- colData$V1

# make variables factors and specify level
cols <- c("sex", "treatment", "tissue")
colData[cols] <- lapply(colData[cols], factor) 
colData$tissue <- factor(colData$tissue, 
                         levels = c("hypothalamus", "pituitary", "gonad"))

# create a combinatorial variable
colData$group <- paste(colData$sex, colData$tissue, colData$treatment, sep=".")

# add variable to distinguish characterization study and manipulation
colData$study <- ifelse(grepl("m.|extend|prolong", colData$treatment), "manipulation", "charcterization")

colData$study <- factor(colData$study, 
                         levels = c("charcterization", "manipulation"))

## view colData
str(colData)
head(colData, 5)

# check that rownames and colnames match for DESeq
ncol(countData) == nrow(colData)

colData %>% select(sex, tissue, treatment, study)  %>%  summary()

```

## save bird and sample and count data 

```{r}
samples <- colData

birds <- colData %>%
  select(bird, sex, treatment) %>%
  distinct()
birds

```



## subset for manipulation study

```{r manip}
# create a new dfs for just the manipulation study
colData_manipluation <- colData %>%
  dplyr::filter(grepl('m.|extend|prolong', treatment)) %>%
  droplevels()
row.names(colData_manipluation) <- colData_manipluation$V1

# print sample sizes
colData_manipluation %>% select(sex,treatment, tissue)  %>%  summary()

# create list with filenames and select countData columns matching the file names
savecols <- as.character(colData_manipluation$V1) 
savecols <- as.vector(savecols) 
countData_manipluation <- countData %>% dplyr::select(one_of(savecols)) 

# confirm that row and col lenghts are equal
ncol(countData_manipluation) == nrow(colData_manipluation)
```

## subset for characterization study

```{r characterization}
# filter out data related to the manipulation study
colData_characterization <- colData %>%
  dplyr::filter(!grepl('m.|extend|prolong', treatment)) %>%
  droplevels()
row.names(colData_characterization) <- colData_characterization$V1

# print sample sizes
colData_characterization %>% select(sex,treatment, tissue)  %>%  summary()

savecols <- as.character(colData_characterization$V1) 
savecols <- as.vector(savecols) 
countData_characterization <- countData %>% dplyr::select(one_of(savecols)) 

# check that row and col lenghts are equal
ncol(countData_characterization) == nrow(colData_characterization)
```

## save files for downstream use

```{r write}
write.csv(countData, "../results/00_counts.csv")
write.csv(samples, "../metadata/00_samples.csv")
write.csv(birds, "../metadata/00_birds.csv")

write.csv(colData_manipluation, "../metadata/00_colData_manipluation.csv", row.names = TRUE) 
write.csv(countData_manipluation, "../results/00_countData_manipluation.csv", row.names = TRUE) 
write.csv(colData_characterization, "../metadata/00_colData_characterization.csv", row.names = TRUE) 
write.csv(countData_characterization, "../results/00_countData_characterization.csv", row.names = TRUE) 

write.csv(geneinfo, "../metadata/00_geneinfo.csv", row.names = TRUE)

write.csv(isoforms, "../results/00_geneswithisoformss.csv", row.names = T)
```


