---
title: "02_DESeq2"
output: md_document
---

```{r setup}
library(tidyverse)
library(DESeq2)
library(BiocParallel)
register(MulticoreParam(6))
library(caret)  # for sampling data
library(cowplot)

source("../R/functions.R")  # load custom functions 
source("../R/themes.R")  # load custom themes and color palletes

knitr::opts_chunk$set(fig.path = '../figures/DESeq2/',  message=F, comment=F, warning=F)
```

## Notes


- DESeq2 was not designed to run on 100+ samples. But, I really like it, so I do it anyways. Some of these commands take like 15 min to run using 6 cores. 
_ Also, I don't run every chuck every time. When I want new analysese, I add them to the bottom and set `eval = T` for chunks I don't want to rerurn 

## Characterization 

```{r readfiles, eval=T}
# import "colData" which contains sample information and "countData" which contains read counts
countData <- read.csv("../results/00_countData_characterization.csv", header = T, row.names = 1)
geneinfo <- read.csv("../metadata/00_geneinfo.csv", row.names = 1)
head(geneinfo)

# craete variable that will be critical for subset later on
colData <- read.csv("../metadata/00_colData_characterization.csv", header = T, row.names = 1)
colData$sextissue <- as.factor(paste(colData$sex, colData$tissue, sep = "_"))
colData$treatment <- factor(colData$treatment, levels = charlevels)
colData$tissue <- factor(colData$tissue, levels = tissuelevel)
levels(colData$treatment)
levels(colData$sex)
levels(colData$sextissue)
levels(colData$tissue)
```



```{r deseq2, eval=F, include=F}
for(i in levels(colData$sextissue)){
  
  newcolData <- subsetcolData2(colData, i)
  
  # save counts that match colData
  savecols <- as.character(newcolData$V1) 
  savecols <- as.vector(savecols) 
  
  newcountData <- countData %>% dplyr::select(one_of(savecols)) 
  
  dds <- DESeqDataSetFromMatrix(countData = newcountData,
                                colData = newcolData,
                                design = ~ treatment )
  dds <- dds[rowSums(counts(dds) > 1) >= 10]  # filter more than sample with less 0 counts
  print(dds)
  print(dim(dds))
  dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
  
  vsd <- as.data.frame(assay(vst(dds, blind=FALSE)))
  
  myfilename = paste0("../results/DEseq2/", i, "_vsd.csv")
  write.csv(vsd, myfilename)
  
  #return(dds)
  #return(vsd)
  print(head(vsd))

  # save differential gene expression results
  control.bldg <- createDEGdfsave("bldg", "control", i)
  bldg.lay <- createDEGdfsave("lay", "bldg", i)
  lay.inc.d3 <- createDEGdfsave("inc.d3", "lay",  i) 
  inc.d3.inc.d9 <- createDEGdfsave("inc.d9", "inc.d3", i) 
  inc.d9.inc.d17 <- createDEGdfsave("inc.d17", "inc.d9", i)
  inc.d17.hatch <- createDEGdfsave( "hatch", "inc.d17", i) 
  hatch.n5 <- createDEGdfsave( "n5", "hatch",  i) 
  n5.n9 <- createDEGdfsave("n9", "n5",  i) 
}

```

## Manipulation 

### Data wrangle

```{r manipulation1, eval = F, include = F}
# import "colData" which contains sample information and "countData" which contains read counts
countData <- read.csv("../results/00_counts.csv", header = T, row.names = 1)
geneinfo <- read.csv("../metadata/00_geneinfo.csv", row.names = 1)
head(geneinfo)


levelsreplace <- c( "m.inc.d8" , "prolong" , "extend")
levelsremoval <- c( "m.inc.d3" ,    "m.inc.d9" , "m.inc.d17" , "m.n2")
controlsremovalreplace <- c( "inc.d3" ,    "inc.d9" , "inc.d17" , "hatch")

manipulationsandcontrols <- c(controlsremoval, levelsreplace, levelsremoval)

colData <- read.csv("../metadata/00_samples.csv", header = T, row.names = 1)
colData$sextissue <- as.factor(paste(colData$sex, colData$tissue, sep = "_"))
colData$treatment <- factor(colData$treatment, levels = manipulationsandcontrols)
colData$tissue <- factor(colData$tissue, levels = tissuelevel)
levels(colData$treatment)
levels(colData$sex)
levels(colData$sextissue)
levels(colData$tissue)
colData <- colData %>% drop_na()
```
### Manipuation: comparisons to respective controls

```{r manipulationcontrol, eval = F}

createDEGdfsaveManip <- function(up, down, mytissue){
  
  res <- results(dds, contrast = c("treatment", up, down), independentFiltering = T, alpha = 0.1)
  
  DEGs <- data.frame(gene = row.names(res),
                        padj = res$padj, 
                        logpadj = -log10(res$padj),
                        lfc = res$log2FoldChange,
                        sextissue = mytissue)
  DEGs <- na.omit(DEGs)
  DEGs <- DEGs %>%
    dplyr::mutate(direction = ifelse(DEGs$lfc > 0 & DEGs$padj < 0.1, 
                                     yes = up, no = ifelse(DEGs$lfc < 0 & DEGs$padj < 0.1, 
                                                           yes = down, no = "NS"))) %>% 
    dplyr::arrange(desc(lfc)) 
  
  DEGs$direction <- factor(DEGs$direction, levels = c(down, "NS", up)) 
  
  # write DEGsframe of only significant genes
  DEGs <- DEGs %>% dplyr::filter(direction != "NS")
  print(str(DEGs))
  
  partialfilename = paste("_", down, "_", up, sep = "")
  myfilename = paste0("../results/DESeq2/manip/", mytissue, partialfilename, "_DEGs.csv")
  
  write.csv(DEGs, myfilename, row.names = F)
  # return DEGs frome with all data, included NS genes
  #print(head(DEGs))
}  

for(i in levels(colData$sextissue)){
  
  newcolData <- subsetcolData2(colData, i)
  
  # save counts that match colData
  savecols <- as.character(newcolData$V1) 
  savecols <- as.vector(savecols) 
  
  newcountData <- countData %>% dplyr::select(one_of(savecols)) 
  
  dds <- DESeqDataSetFromMatrix(countData = newcountData,
                                colData = newcolData,
                                design = ~ treatment )
  dds <- dds[rowSums(counts(dds) > 1) >= 10]  # filter more than sample with less 0 counts
  print(dds)
  print(dim(dds))
  dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
  
  vsd <- as.data.frame(assay(vst(dds, blind=FALSE)))
  
  myfilename = paste0("../results/DEseq2/manip/", i, "_vsd.csv")
  write.csv(vsd, myfilename)
  
  #return(dds)
  #return(vsd)
  #print(head(vsd))

  # save differential gene expression results
  inc.d3.m.inc.d3 <- createDEGdfsaveManip("m.inc.d3", "inc.d3",  i) 
  inc.d9.m.inc.d9 <- createDEGdfsaveManip("m.inc.d9", "inc.d9",  i) 
  inc.d17.m.inc.d17 <- createDEGdfsaveManip("m.inc.d17", "inc.d17",  i) 
  hatch.m.n2 <- createDEGdfsaveManip("m.n2", "hatch",  i) 
  inc.d9.m.inc.d8 <- createDEGdfsaveManip("m.inc.d8", "inc.d9",  i) 
  inc.d17.prolong <- createDEGdfsaveManip("prolong", "inc.d17",  i) 
  hatch.extend <- createDEGdfsaveManip("extend", "hatch",  i) 
}

```


## Manipulation: comparisons to other manipluations

```{r manipulationwithin, eval = F}
# import "colData" which contains sample information and "countData" which contains read counts
countData <- read.csv("../results/00_counts.csv", header = T, row.names = 1)
geneinfo <- read.csv("../metadata/00_geneinfo.csv", row.names = 1)
head(geneinfo)


levelsreplace <- c( "m.inc.d8" , "prolong" , "extend")
levelsremoval <- c( "m.inc.d3" ,    "m.inc.d9" , "m.inc.d17" , "m.n2")
controlsremovalreplace <- c( "inc.d3" ,    "inc.d9" , "inc.d17" , "hatch")

manipulationsandcontrols <- c(controlsremoval, levelsreplace, levelsremoval)

colData <- read.csv("../metadata/00_samples.csv", header = T, row.names = 1)
colData$sextissue <- as.factor(paste(colData$sex, colData$tissue, sep = "_"))
colData$treatment <- factor(colData$treatment, levels = manipulationsandcontrols)
colData$tissue <- factor(colData$tissue, levels = tissuelevel)
levels(colData$treatment)
levels(colData$sex)
levels(colData$sextissue)
levels(colData$tissue)
colData <- colData %>% drop_na()

createDEGdfsaveManip <- function(up, down, mytissue){
  
  res <- results(dds, contrast = c("treatment", up, down), independentFiltering = T, alpha = 0.1)
  
  DEGs <- data.frame(gene = row.names(res),
                        padj = res$padj, 
                        logpadj = -log10(res$padj),
                        lfc = res$log2FoldChange,
                        sextissue = mytissue)
  DEGs <- na.omit(DEGs)
  DEGs <- DEGs %>%
    dplyr::mutate(direction = ifelse(DEGs$lfc > 0 & DEGs$padj < 0.1, 
                                     yes = up, no = ifelse(DEGs$lfc < 0 & DEGs$padj < 0.1, 
                                                           yes = down, no = "NS"))) %>% 
    dplyr::arrange(desc(lfc)) 
  
  DEGs$direction <- factor(DEGs$direction, levels = c(down, "NS", up)) 
  
  # write DEGsframe of only significant genes
  DEGs <- DEGs %>% dplyr::filter(direction != "NS")
  print(str(DEGs))
  
  partialfilename = paste("_", down, "_", up, sep = "")
  myfilename = paste0("../results/DESeq2/manip/", mytissue, partialfilename, "_DEGs.csv")
  
  write.csv(DEGs, myfilename, row.names = F)
  # return DEGs frome with all data, included NS genes
  #print(head(DEGs))
}  


for(i in levels(colData$sextissue)){
  
  newcolData <- subsetcolData2(colData, i)
  
  # save counts that match colData
  savecols <- as.character(newcolData$V1) 
  savecols <- as.vector(savecols) 
  
  newcountData <- countData %>% dplyr::select(one_of(savecols)) 
  
  dds <- DESeqDataSetFromMatrix(countData = newcountData,
                                colData = newcolData,
                                design = ~ treatment )
  dds <- dds[rowSums(counts(dds) > 1) >= 10]  # filter more than sample with less 0 counts
  print(dds)
  print(dim(dds))
  dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
  
  vsd <- as.data.frame(assay(vst(dds, blind=FALSE)))
  
  myfilename = paste0("../results/DEseq2/manip/", i, "_vsd.csv")
  write.csv(vsd, myfilename)
  
  #return(dds)
  #return(vsd)
  #print(head(vsd))

  # save differential gene expression results
  hatch.m.n2 <- createDEGdfsaveManip("m.n2", "hatch",  i) 
  prolong.extend <- createDEGdfsaveManip("extend", "prolong",  i) 
  early.extend <- createDEGdfsaveManip("extend", "m.inc.d8",  i) 
  early.prolong <- createDEGdfsaveManip("prolong", "m.inc.d8",  i) 
  m.inc.d9.m.inc.d3 <- createDEGdfsaveManip("m.inc.d9", "m.inc.d3",  i) 
  m.inc.d17.m.inc.d9 <- createDEGdfsaveManip("m.inc.d17", "m.inc.d9",  i) 
  m.inc.d17.m.inc.d3 <- createDEGdfsaveManip("m.inc.d17", "m.inc.d3",  i) 
  m.n2.m.inc.d17 <- createDEGdfsaveManip("m.n2", "m.inc.d17",  i) 
  m.n2.m.inc.d9 <- createDEGdfsaveManip("m.n2", "m.inc.d9",  i)
  m.n2.m.inc.d3 <- createDEGdfsaveManip("m.n2", "m.inc.d3",  i)
  prolong.m.inc.d17 <- createDEGdfsaveManip("prolong", "m.inc.d17",  i) 
  m.inc.d8.m.inc.d9 <- createDEGdfsaveManip("m.inc.d8", "m.inc.d9",  i) 
  extend.m.n2 <- createDEGdfsaveManip("extend", "m.n2",  i) 
}

```


## PRL-driven DESeq2

```{r PRL, eval = F}
PRLdata <- read_csv("../results/PRLvsd.csv")

PRLsamples <- colData %>%
  mutate(samples = V1)  %>%
  select(samples) 
PRLsamples

# join with hi lo, and drop controls
colDataPRL <- full_join(PRLdata, PRLsamples) %>% 
  drop_na() %>%
  select(samples, tissue, sex, treatment, hiloPRL) %>% 
  filter(!treatment %in% c("control", "bldg"))

# sample 100 only, better for deseq2
colDataPRL <- colDataPRL[sample(1:nrow(colDataPRL), 100,
   replace=FALSE),]

colDataPRL <- as.data.frame(colDataPRL)
row.names(colDataPRL) <- colDataPRL$samples

savecols <- as.character(colDataPRL$samples) 
savecols <- as.vector(savecols) 
  
newcountData <- countData %>% dplyr::select(one_of(savecols)) 

colDataPRL %>%
  group_by(sex, tissue, hiloPRL) %>%
  summarize(n = n())
  
dds <- DESeqDataSetFromMatrix(countData = newcountData,
                                colData = colDataPRL,
                                design = ~sex * hiloPRL )
dds <- dds[rowSums(counts(dds) > 1) >= 10]  # filter more than sample with less 0 counts
print(dds)
print(dim(dds))
dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
  
vsd <- as.data.frame(assay(vst(dds, blind=FALSE)))

ressex <- results( dds, contrast = c("sex", "male", "female") )
reshiloPRL <- results( dds, contrast = c("hiloPRL", "hi", "lo") )
hist( ressex$pvalue, breaks=20, col="grey" )
hist( reshiloPRL$pvalue, breaks=20, col="grey" )

restoDEGs <- function(res, up, down){
  DEGs <- data.frame(gene = row.names(res),
                        padj = res$padj, 
                        logpadj = -log10(res$padj),
                        lfc = res$log2FoldChange)
  DEGs <- na.omit(DEGs)
  DEGs <- DEGs %>%
    dplyr::mutate(direction = ifelse(DEGs$lfc > 0 & DEGs$padj < 0.1, 
                                     yes = up, no = ifelse(DEGs$lfc < 0 & DEGs$padj < 0.1, 
                                                           yes = down, no = "NS"))) %>% 
    dplyr::arrange(gene)
  DEGs$direction <- factor(DEGs$direction, levels = c(down, "NS", up)) 
  DEGs <- DEGs %>% dplyr::filter(direction != "NS")
  print(head(DEGs))
  return(DEGs)
}  

sexdegs <- restoDEGs(ressex, "male", "female")
PRLdegs <- restoDEGs(reshiloPRL, "hi", "lo")

plotbarvolcano <- function(df, mylabels, mycolorname){
  p1 <- ggplot(df, aes(x = lfc, y = logpadj, color = direction)) +
    geom_point() +
    scale_color_manual(values = allcolors, name = mycolorname) +
    theme_B3() +
    theme(legend.position = "top") +
    labs(x = "Log fold change", y = "Log adj. p-value") +
    xlim(-5,5)

  p2 <- ggplot(df, aes(x = direction,  fill = direction)) +
    geom_bar() +
    theme_B3() +
    scale_fill_manual(values = allcolors) +
    theme_B3() +
    theme(legend.position = "none") +
    labs(x = mycolorname, y = "Total DEGs")
  
  p <- plot_grid(p1,p2, labels  = mylabels, label_size = 12)
  plot(p)
}

top <- plotbarvolcano(sexdegs, c("a","b"), "sex")
bottom <- plotbarvolcano(PRLdegs, c("c","d"), "PRL")

ab <- plot_grid(top,bottom, nrow = 2)

sexPRLdesg <- full_join(sexdegs, PRLdegs, by = "gene")

sexPRLdesg$direction.x <- as.character(sexPRLdesg$direction.x)

sexPRLdesg <- full_join(sexdegs, PRLdegs, by = "gene") %>%
  dplyr::select(gene, direction.x, direction.y) %>%
  tidyr::replace_na(list(direction.x = "no sex effect", direction.y = "no PRL effect"))
sexPRLdesg

write.csv(sexPRLdesg, "../results/DEG-PRLsex.csv")

sexPRLdesg2 <- sexPRLdesg %>%
  mutate(higherin = paste(direction.x, direction.y, sep = "_")) %>%
  mutate(direction = paste(higherin, "PRL", sep = "")) %>%
  select(gene, direction) %>%
  group_by(direction) %>%
  summarize(DEGs = str_c(gene , collapse = " "))
sexPRLdesg2
write.csv(sexPRLdesg2, "../results/PRLsexinteraction.csv")

c <- ggplot(sexPRLdesg, aes(x = direction.x, fill = direction.y)) +
  geom_bar(position = "dodge") +
  scale_fill_manual(values = allcolors, na.value= "#bdbdbd",
                    labels= c("lo", "hi", "NS"),
                    name = "PRL") +
  scale_x_discrete(breaks = c("female", "male", NA),
                   labels = c("female", "male", "NS"),
                   name = "Sex")  +
  labs(y = "Total DEGs") +
  theme_B3() +
  theme(legend.position = "top")

plot_grid(ab, c, nrow = 1, rel_widths = c(2,1.2), labels = c(" ", "d"), label_size = 12)
```



## Hypothesis testing - external stimuli and/or prolactin

```{r, eval = F}
# label each bird with weather or not their pituitary had hi PRL or not
# then make new column with nest, eggs, or chicks, drop controls

PRLdata <- read_csv("../results/PRLvsd.csv") %>%
  mutate(V1 = samples) %>%
  mutate(bird = sapply(strsplit(as.character(samples),'\\_'), "[", 1)) %>%
  select(bird,hiloPRL)

# join with hi lo, and drop controls
colDatahypothesis <- full_join(colData, PRLdata, by = "bird") %>% 
  drop_na() %>%
  select(V1, tissue, sex, sextissue, treatment, hiloPRL) %>% 
  filter(!treatment %in% c("control", "bldg")) %>%
  mutate(external = fct_collapse(treatment, eggs = c("lay", "inc.d3", "inc.d9", "inc.d17"),
                                            chicks = c("hatch", "n5", "n9")))

colDatahypothesis <- as.data.frame(colDatahypothesis)
row.names(colDatahypothesis) <- colDatahypothesis$V1

savecols <- as.character(colDatahypothesis$V1) 
savecols <- as.vector(savecols) 
  
newcountData <- countData %>% dplyr::select(one_of(savecols)) 

colDatahypothesis %>%
  group_by(sex, tissue, hiloPRL, external) %>%
  summarize(n = n())

colDatahypothesis %>%
  filter(tissue == "pituitary")  %>%
  group_by(sex, tissue, hiloPRL, treatment) %>%
  summarize(n = n())


hypothesesistesting <- function(whichfactor, up, down, mytissue){
  
  res <- results(dds, contrast = c(whichfactor, up, down), independentFiltering = T, alpha = 0.1)
  
  DEGs <- data.frame(gene = row.names(res),
                        padj = res$padj, 
                        logpadj = -log10(res$padj),
                        lfc = res$log2FoldChange,
                        sextissue = mytissue)
  DEGs <- na.omit(DEGs)
  DEGs <- DEGs %>%
    dplyr::mutate(direction = ifelse(DEGs$lfc > 0 & DEGs$padj < 0.1, 
                                     yes = up, no = ifelse(DEGs$lfc < 0 & DEGs$padj < 0.1, 
                                                           yes = down, no = "NS"))) %>% 
    dplyr::arrange(desc(lfc)) 
  
  DEGs$direction <- factor(DEGs$direction, levels = c(down, "NS", up)) 
  
  # write DEGsframe of only significant genes
  DEGs <- DEGs %>% dplyr::filter(direction != "NS")
  print(str(DEGs))
  
  partialfilename = paste("_", down, "_", up, sep = "")
  myfilename = paste0("../results/DESeq2/hypothesis/", mytissue, partialfilename, "_DEGs.csv")
  
  write.csv(DEGs, myfilename, row.names = F)
  # return DEGs frome with all data, included NS genes
  #print(head(DEGs))
}  

for(i in levels(colDatahypothesis$sextissue)){
  
  newcolData <- subsetcolData2(colDatahypothesis, i)
  
  # save counts that match colData
  savecols <- as.character(newcolData$V1) 
  savecols <- as.vector(savecols) 
  
  newcountData <- countData %>% dplyr::select(one_of(savecols)) 
  
  dds <- DESeqDataSetFromMatrix(countData = newcountData,
                                colData = newcolData,
                                design = ~ external + hiloPRL )
  dds <- dds[rowSums(counts(dds) > 1) >= 10]  # filter more than sample with less 0 counts
  print(dds)
  print(dim(dds))
  dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
  
  vsd <- as.data.frame(assay(vst(dds, blind=FALSE)))
  
  myfilename = paste0("../results/DEseq2/hypothesis/", i, "_vsd.csv")
  write.csv(vsd, myfilename)

  # save differential gene expression results
  # sexmalefemale <- hypothesesistesting("sex", "male", "female",  i) 
  externalschickseggs <- hypothesesistesting("external", "chicks", "eggs",  i) 
  hiloPRL <- hypothesesistesting("hiloPRL", "hi","lo",  i) 
}
```


## tissue and sex differences


```{r eval = F}

## first or sex

for(i in levels(colData$tissue)){
  
  newcolData <- subsetcolData4(colData, i)
  print(head(newcolData))
  # save counts that match colData
  savecols <- as.character(newcolData$V1) 
  savecols <- as.vector(savecols) 
  
  newcountData <- countData %>% dplyr::select(one_of(savecols)) 
  
  dds <- DESeqDataSetFromMatrix(countData = newcountData,
                                colData = newcolData,
                                design = ~ sex )

  
  dds <- dds[rowSums(counts(dds) > 1) >= 10]  # filter more than sample with less 0 counts
  print(dds)
  print(dim(dds))
  dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
  
  vsd <- as.data.frame(assay(vst(dds, blind=FALSE)))
  
  myfilename = paste0("../results/DEseq2/sex/", i, "_vsd.csv")
  write.csv(vsd, myfilename)

  print(head(vsd))

  # save differential gene expression results
  sex <- createDEGdfsavesex("male", "female", i)
}


## then for tissue

dds <- DESeqDataSetFromMatrix(countData = countData,
                                colData = colData,
                                design = ~ tissue )
dds <- dds[rowSums(counts(dds) > 1) >= 10]  # filter more than sample with less 0 counts
print(dds)
print(dim(dds))
dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
  
vsd <- as.data.frame(assay(vst(dds, blind=FALSE)))
write.csv(vsd, "../results/DEseq2/tissue/tissuevsds.csv")
print(head(vsd))

# save differential gene expression results
HP <- createDEGdfsavestissue("pituitary", "hypothalamus")
HG <- createDEGdfsavestissue("gonad", "hypothalamus")
PG <- createDEGdfsavestissue("gonad", "pituitary")
  
```







