---
title: "06_DESeq2"
output: md_document
---

```{r setup, include = FALSE}

library(tidyverse)
library(DESeq2)

library(BiocParallel)
register(MulticoreParam(6))

source("../R/functions.R")  # load custom functions 
source("../R/themes.R")  # load custom themes and color palletes
source("../R/icons.R")  # for images

knitr::opts_chunk$set(fig.path = '../figures/DESeq2/',  message=F, comment=F, warning=F)
```

# Warning: This script takes a very long time to run. 

DESeq2 was not designed to run on 300 samples. But, I really like it, so I do it anyways. Some of these commands take like 15 min to run using 6 cores. 

## DEseq2 on all chaacterization and manipulations

```{r readfiles}
# import "colData" which contains sample information and "countData" which contains read counts
countData <- read.csv("../results/00_countData_characterization.csv", header = T, row.names = 1)
geneinfo <- read.csv("../metadata/00_geneinfo.csv", row.names = 1)
head(geneinfo)

# craete variable that will be critical for subset later on
colData <- read.csv("../metadata/00_colData_characterization.csv", header = T, row.names = 1)
colData$sextissue <- as.factor(paste(colData$sex, colData$tissue, sep = "_"))
colData$treatment <- factor(colData$treatment, levels = charlevels)
colData$tissue <- factor(colData$tissue, levels = tissuelevel)
levels(colData$treatment)
levels(colData$sex)
levels(colData$sextissue)
levels(colData$tissue)
```



```{r deseq2, eval=T}
for(i in levels(colData$sextissue)){
  
  newcolData <- subsetcolData2(colData, i)
  
  # save counts that match colData
  savecols <- as.character(newcolData$V1) 
  savecols <- as.vector(savecols) 
  
  newcountData <- countData %>% dplyr::select(one_of(savecols)) 
  
  dds <- DESeqDataSetFromMatrix(countData = newcountData,
                                colData = newcolData,
                                design = ~ treatment )
  dds <- dds[rowSums(counts(dds) > 1) >= 10]  # filter more than sample with less 0 counts
  print(dds)
  print(dim(dds))
  dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
  
  vsd <- as.data.frame(assay(vst(dds, blind=FALSE)))
  
  myfilename = paste0("../results/DEseq2/", i, "_vsd.csv")
  write.csv(vsd, myfilename)
  
  #return(dds)
  #return(vsd)
  print(head(vsd))

  # save differential gene expression results
  control.bldg <- createDEGdfsave("bldg", "control", i)
  bldg.lay <- createDEGdfsave("lay", "bldg", i)
  lay.inc.d3 <- createDEGdfsave("inc.d3", "lay",  i) 
  inc.d3.inc.d9 <- createDEGdfsave("inc.d9", "inc.d3", i) 
  inc.d9.inc.d17 <- createDEGdfsave("inc.d17", "inc.d9", i)
  inc.d17.hatch <- createDEGdfsave( "hatch", "inc.d17", i) 
  hatch.n5 <- createDEGdfsave( "n5", "hatch",  i) 
  n5.n9 <- createDEGdfsave("n9", "n5",  i) 
}

```

