---
title: "02_DESeq_characterization"
output: md_document
---

```{r setup, message=F, comment=FALSE, warning=FALSE, results='hide'}
library(tidyverse)
library(DESeq2)
library(cowplot)
library(RColorBrewer)
library(pheatmap)
library(kableExtra)
library(viridis)
library(ggimage)


# load custom functions  
source("../R/functions.R") 

knitr::opts_chunk$set(fig.path = '../figures/characterization/', cache = TRUE)
```

## Characterization data

```{r readfiles}
# import "colData" which contains sample information and "countData" which contains read counts
c.colData <- read.csv("../metadata/00_colData_characterization.csv", header = T, row.names = 1)
c.countData <- read.csv("../results/00_countData_characterization.csv", header = T, row.names = 1)
geneinfo <- read.csv("../metadata/00_geneinfo.csv", row.names = 1)

# set levels
c.colData$treatment <- factor(c.colData$treatment, levels = 
                              c("control",  "bldg", "lay", "inc.d3", "inc.d9", "inc.d17", "hatch", "n5", "n9"))
levels(c.colData$treatment)
c.colData$sextissue <- as.factor(paste(c.colData$sex, c.colData$tissue, sep = "_"))
summary(c.colData[c(7,3,4,5,8)])
```



## Run DESeq on all subsets of the data

```{r dds, cache = T, eval = T}
dds.female_hypothalamus <- subsetDESeq(c.colData,  c.countData, "female_hypothalamus")
dds.female_pituitary <- subsetDESeq(c.colData,  c.countData, "female_pituitary" )
dds.female_gonad <- subsetDESeq(c.colData,  c.countData, "female_gonad" )
dds.male_hypothalamus <- subsetDESeq(c.colData,  c.countData, "male_hypothalamus" )
dds.male_pituitary <- subsetDESeq(c.colData,  c.countData, "male_pituitary"  )
dds.male_gondad <- subsetDESeq(c.colData,  c.countData, "male_gonad")
```

## Calculate and plot total DEGs


```{r returntotalDEGs, eval = T, results = 'hide'}
DEGs.female_hypothalamus <- returntotalDEGs(dds.female_hypothalamus)
DEGs.female_pituitary <- returntotalDEGs(dds.female_pituitary)
DEGs.female_gonad <- returntotalDEGs(dds.female_gonad)
DEGs.male_hypothalamus <- returntotalDEGs(dds.male_hypothalamus)
DEGs.male_pituitary <- returntotalDEGs(dds.male_pituitary)
DEGs.male_gondad <- returntotalDEGs(dds.male_gondad)
```

```{r totalDEGs, eval = T}
a <- plottotalDEGs(DEGs.female_hypothalamus, "female hypothalamus")
b <- plottotalDEGs(DEGs.female_pituitary, "female pituitary")
c <- plottotalDEGs(DEGs.female_gonad, "female gonad")
d <- plottotalDEGs(DEGs.male_hypothalamus, "male hypothalamus")
e <- plottotalDEGs(DEGs.male_pituitary, "male pituitary")
f <- plottotalDEGs(DEGs.male_gondad, "male gonad")

plot_grid(a + theme(legend.position = "none"),
          b + theme(legend.position = "none"),
          c,
          d + theme(legend.position = "none"),
          e + theme(legend.position = "none"),
          f,
          nrow = 2, rel_widths = c(0.3, 0.3, 0.4)) 
```



## Plot only a subset of DEGs that relate to hypotheses as bar and line graphs

```{r barplots}
# bars with lines
g <- plotserialDEGs(DEGs.female_hypothalamus, "Female hypothalamus", "#00A08A")
h <- plotserialDEGs(DEGs.female_pituitary, "Female pituitary", "#00A08A")
i <- plotserialDEGs(DEGs.female_gonad, "Female gonad", "#00A08A")
j <- plotserialDEGs(DEGs.male_hypothalamus, "Male hypothalamus", "#F98400")
k <- plotserialDEGs(DEGs.male_pituitary, "Male pituitary", "#F98400")
l <- plotserialDEGs(DEGs.male_gondad, "Male gonad", "#F98400")

mybarplots <- plot_grid(g, h , i , j ,k , l, nrow = 2) 
mybarplots

pdf("../figures/characterization/barplots.pdf", width = 10, height = 6)
plot(mybarplots)
dev.off()
```


```{r linegraphs}

# lines only, different colors for different sex tissue groups
FH <- subsetDEGs(DEGs.female_hypothalamus, "Female hypothalamus")
FP <- subsetDEGs(DEGs.female_pituitary, "Female pituitary")
FG <- subsetDEGs(DEGs.female_gonad, "Female gonad")
MH <- subsetDEGs(DEGs.male_hypothalamus, "Male hypothalamus")
MP <- subsetDEGs(DEGs.male_pituitary, "Male pituitary")
MG <- subsetDEGs(DEGs.male_gondad, "Male gonad")

specificDEGs <- rbind(FH, FP, FG, MH, MP, MG)

specificDEGs$sextissue <- factor(specificDEGs$sextissue, levels = c("Female hypothalamus", "Female pituitary", "Female gonad",
                                                                    "Male hypothalamus", "Male pituitary", "Male gonad"))
mylinegraph <- ggplot(specificDEGs, aes(x=comparison, y=V3, group = sextissue)) +
    geom_line(aes(color=sextissue)) +
    geom_point(aes(color=sextissue)) + 
    theme_minimal(base_size = 10) +
    #theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    labs(x = "Parental stage transitions", y= "No. of differentially expressed genes ",
         title = "Gene expression changes during parental care transitions") + 
    scale_x_discrete(labels=c("Control to\nNest bldg", "Nest bldg. to\nLay", "Lay to\nInc. day 3",
                              "Inc. day 3 to\nInc. day 9", "Inc. day 9 to\nInc. day 17",
                              "Inc. day 17 to\nHatch", "Hatch to\nNestling day 5", "Nestling day 5 to\n Nestling day 9")) +
   theme(legend.title = element_blank(),
         legend.position = c(0.8, 0.8))
mylinegraph

pdf("../figures/characterization/linegraphs.pdf", width = 8, height = 4)
plot(mylinegraph)
dev.off()
```


## Calculate and plot principal component

```{r pca, eval = T, cache = TRUE}

FhypPCA <- returnPCAs(dds.female_hypothalamus)
plotPC12(FhypPCA, "female hypothalamus")

FpitPCA <- returnPCAs(dds.female_pituitary)      
plotPC12(FpitPCA, "female pituitary")

FgonPCA <- returnPCAs(dds.female_gonad)
plotPC12(FgonPCA, "female gonad")

MhypPCA <- returnPCAs(dds.male_hypothalamus)
plotPC12(MhypPCA, "male hypothalamus")

MpitPCA <- returnPCAs(dds.male_pituitary)
plotPC12(MpitPCA, "male pituitary")

MgonPCA <- returnPCAs(dds.male_gondad)
plotPC12(MgonPCA, "male gonad")
```

## heamap of candidate genes

```{r pheatmap, eval = T, cache = TRUE}
colDataFH <- subsetcolData(c.colData, "female_hypothalamus")
colDataFP <- subsetcolData(c.colData, "female_pituitary")
colDataFG <- subsetcolData(c.colData, "female_gonad")
colDataMH <- subsetcolData(c.colData, "male_hypothalamus")
colDataMP <- subsetcolData(c.colData, "male_pituitary")
colDataMG <- subsetcolData(c.colData, "male_gonad")


makepheatmap(dds.female_hypothalamus, colDataFH, "female hypothalamus")
makepheatmap(dds.female_pituitary, colDataFP, "female pituitary")
makepheatmap(dds.female_gonad, colDataFG, "female gonad")
makepheatmap(dds.male_hypothalamus, colDataMH,  "male hypothalamus")
makepheatmap(dds.male_pituitary, colDataMP, "male pituitary")
makepheatmap(dds.male_gondad, colDataMG, "male gonad")
```
## plot candidate genes

```{r candidates, eval = T}
plotcandidates(dds.female_hypothalamus, colDataFH,"female hypothalamus")
plotcandidates(dds.female_pituitary, colDataFP, "female pituitary")
plotcandidates(dds.female_gonad, colDataFG,"female gonad")
plotcandidates(dds.male_hypothalamus, colDataMH,"male hypothalamus")
plotcandidates(dds.male_pituitary,colDataMP, "male pituitary")
plotcandidates(dds.male_gondad,colDataMG, "male gonad")
```

## plot prolactin

```{r prolactin, eval = T}
a <- plotprolactin(dds.female_hypothalamus, colDataFH,"female hypothalamus")
b <- plotprolactin(dds.female_pituitary, colDataFP, "female pituitary")
c <- plotprolactin(dds.female_gonad, colDataFG,"female gonad")
d <- plotprolactin(dds.male_hypothalamus, colDataMH,"male hypothalamus")
e <- plotprolactin(dds.male_pituitary,colDataMP, "male pituitary")
f <- plotprolactin(dds.male_gondad,colDataMG, "male gonad")

mylegend <- get_legend(f)

plots <- plot_grid(a + theme(legend.position = "none"),
          b + theme(legend.position = "none"),
          c + theme(legend.position = "none"),
          d + theme(legend.position = "none"), 
          e + theme(legend.position = "none"),
          f + theme(legend.position = "none"))
plot_grid(plots, mylegend, nrow = 2, rel_heights = c(1,0.1))

```

## correlation heat maps

```{r correlationheatmaps, eval = F}
plotcorrelationheatmaps(dds.female_hypothalamus, colData.female_hypothalamus ,"female hypothalamus: hatch, nest bldg., lay and day-before-hatch cluster")
plotcorrelationheatmaps(dds.female_pituitary, colData.female_pituitary, "female pituitary: clustering suggests an internal clock effect")
plotcorrelationheatmaps(dds.female_gonad, colData.female_gonad,  "female gonad: highly correlated across incubation and early nestling care")
plotcorrelationheatmaps(dds.male_hypothalamus, colData.male_hypothalamus, "male hypothalamus: 100% correlated across most parental care stages")
plotcorrelationheatmaps(dds.male_pituitary, colData.male_pituitary, "male pituitary: non-parental males are most different")
plotcorrelationheatmaps(dds.male_gondad, colData.male_gondad, "male gonad: non-parental males are most different")
```
