---
title: "03_wrangle"
output: md_document
---
  

```{r setup}
library(tidyverse)
library(readr)
library(VennDiagram)

source("../R/themes.R")

knitr::opts_chunk$set(echo = TRUE, message = F, fig.path = '../figures/')
```

### candidate gene anlayses

```{r candidategenes}
# all genes in this parental care dataset 
geneids <- read_csv("../metadata/00_geneinfo.csv") %>% select(-X1) 

# genes from Ch 17 Evolution of parental care
curleychampagnegenes <- c("AVP", "AVPR1A", "ADRA2A",    
                          "COMT", "CRH", "CRHBP", "CRHR1", "CRHR2", 
                          "DRD1", "DRD4","ESR1", "ESR2", #ERa ERbeta
                          "FOS", "HTR2C", "MEST", "NR3C1", #GR
                          "OPRM1", "OXT" , "PGR", "PRL", "PRLR",  "SLC6A4") #5HTT

datadrivengenes <- c("COX1", "COX2", "CYTB", "ATP2B4", "LOC107055658", 
                     "CHGA", "TF", "OVALY", "ANXA5")

literaturegenes <- c(curleychampagnegenes, datadrivengenes,
                     "VIP", "GNIH", "GAL", "TH", "THRB",
                     "GNRHR", "GNRH1", "CGNRH-R", "FSHB", "FSHR")
literaturegenes

literaturegenes <- as.data.frame(literaturegenes) %>%
  mutate(GO = "literature", gene = literaturegenes)  %>%
  select(GO, gene)

# candidate gens from parental care GO terms 
GO_path <- "../metadata/goterms/"   # path to the data
GO_files <- dir(GO_path, pattern = "*.txt") # get file names
GO_pathfiles <- paste0(GO_path, GO_files)

GOgenesLong <- GO_pathfiles %>%
  setNames(nm = .) %>% 
  map_df(~read_table(.x, col_types = cols(), col_names = FALSE), .id = "file_name") %>% 
  mutate(GO = sapply(strsplit(as.character(file_name),'../metadata/goterms/'), "[", 2)) %>% 
  mutate(GO = sapply(strsplit(as.character(GO),'.txt'), "[", 1)) %>% 
  mutate(gene = sapply(strsplit(as.character(X1), "[\\\\]|[^[:print:]]" ), "[", 2)) %>% 
  select(GO, gene)  %>%
  filter(gene != "Symbol") %>%
  distinct(GO,gene)  %>%
  mutate(gene = toupper(gene)) %>%
  rbind(., literaturegenes) %>%
  arrange(gene) %>%
  left_join(., geneids, by = "gene") %>%
  arrange(gene) %>%
  drop_na() 
head(GOgenesLong)

parentalcaregenes <- GOgenesLong %>% 
  pivot_wider(
    names_from = GO,
    values_from = gene,
    values_fill = list(gene = "")) %>%
  left_join(geneids, by = c("geneid","NCBI")) %>%
  dplyr::rename("GO"= "parentalbehavior" )
head(parentalcaregenes)

```


### variance stabilized gene expression  (vsd) 

```{r allvsd}
vsd_path <- "../results/DEseq2/treatment/"   # path to the data
vsd_files <- dir(vsd_path, pattern = "*vsd.csv") # get file names
vsd_pathfiles <- paste0(vsd_path, vsd_files)
vsd_files

## before pivoting, check names of df with
## head(names(allvsd)) and tail(names(allvsd))

allvsd <- vsd_pathfiles %>%
  setNames(nm = .) %>% 
  map_df(~read_csv(.x), .id = "file_name")  %>% 
  dplyr::rename("gene" = "X1") %>% 
  #pivot_longer(cols = L.G118_female_gonad_control:y98.o50.x_male_pituitary_inc.d3, 
               #names_to = "samples", values_to = "counts") 
  pivot_longer(cols = L.G118_female_hypothalamus_control.NYNO:y98.g54_female_hypothalamus_m.hatch, 
               names_to = "samples", values_to = "counts") 
allvsd %>% select(-file_name) %>% head()

tail((allvsd))
```



### vsd for all and candidate genes


```{r candidatevsd}
candidategenes <- GOgenesLong %>% distinct(gene) %>% pull(gene)

getcandidatevsd <- function(whichgenes, whichtissue, whichsex){
  candidates  <- allvsd %>%
    filter(gene %in% whichgenes) %>%
    dplyr::mutate(sextissue = sapply(strsplit(file_name, '_vsd.csv'), "[", 1)) %>%
    dplyr::mutate(sextissue = sapply(strsplit(sextissue, '../results/DEseq2/treatment/'), "[", 2)) %>%
    dplyr::mutate(sex = sapply(strsplit(sextissue, '\\_'), "[", 1),
                  tissue = sapply(strsplit(sextissue, '\\_'), "[", 2),
                  treatment = sapply(strsplit(samples, '\\_'), "[", 4)) %>%
    dplyr::mutate(treatment = sapply(strsplit(treatment, '.NYNO'), "[", 1)) %>%
    dplyr::select(sex, tissue, treatment, gene, samples, counts) %>%
    filter(tissue == whichtissue, sex %in% whichsex)  %>%
    drop_na()
  #candidates$treatment <- factor(candidates$treatment, levels = alllevels)
  return(candidates)
}


hypvsd <- getcandidatevsd(candidategenes, "hypothalamus", sexlevels)
pitvsd <- getcandidatevsd(candidategenes, "pituitary", sexlevels)
gonvsd <- getcandidatevsd(candidategenes, "gonad", sexlevels)
candidatevsd <- rbind(hypvsd, pitvsd)
candidatevsd <- rbind(candidatevsd, gonvsd)

head(candidatevsd)

```

## All DEGs 

```{r allDEG, results=F}
DEG_path <- "../results/DEseq2/treatment/"   # path to the data
DEG_files <- dir(DEG_path, pattern = "*DEGs") # get file names
DEG_pathfiles <- paste0(DEG_path, DEG_files)
#DEG_files

allDEG <- DEG_pathfiles %>%
  setNames(nm = .) %>% 
  map_df(~read_csv(.x), .id = "file_name") %>% 
  mutate(DEG = sapply(strsplit(as.character(file_name),'./results/DEseq2/treatment/'), "[", 2))  %>% 
  mutate(DEG = sapply(strsplit(as.character(DEG),'_diffexp.csv'), "[", 1))  %>% 
  mutate(tissue = sapply(strsplit(as.character(DEG),'\\.'), "[", 1)) %>%
  mutate(down = sapply(strsplit(as.character(DEG),'\\_'), "[", 3)) %>%
  mutate(up = sapply(strsplit(as.character(DEG),'\\_'), "[", 4)) %>%
  mutate(comparison = paste(down,up, sep = "_")) %>%
  mutate(sex = sapply(strsplit(as.character(sextissue),'\\_'), "[", 1)) %>%
  mutate(tissue = sapply(strsplit(as.character(sextissue),'\\_'), "[", 2)) %>%
dplyr::select(sex,tissue,comparison, direction, gene, lfc, padj, logpadj) 
head(allDEG)

unique(allDEG$comparison)

tempDEGs <- DEG_pathfiles %>%
  setNames(nm = .) %>% 
  map_df(~read_csv(.x), .id = "file_name") %>% 
  mutate(DEG = sapply(strsplit(as.character(file_name),'./results/DEseq2/treatment/'), "[", 2))  %>% 
  mutate(DEG = sapply(strsplit(as.character(DEG),'_diffexp.csv'), "[", 1))  %>% 
  mutate(tissue = sapply(strsplit(as.character(DEG),'\\.'), "[", 1)) %>%
  mutate(down = sapply(strsplit(as.character(DEG),'\\_'), "[", 3)) %>%
  mutate(up = sapply(strsplit(as.character(DEG),'\\_'), "[", 4)) %>%
  mutate(comparison = paste(down,up, sep = "_")) %>%
  mutate(sex = sapply(strsplit(as.character(sextissue),'\\_'), "[", 1)) %>%
  mutate(tissue = sapply(strsplit(as.character(sextissue),'\\_'), "[", 2))
head(tempDEGs)

wideDEGs <- tempDEGs %>%
  mutate(lfc = round(lfc, 2),
         padj =  round(padj,2)) %>%
  mutate(sextissue = paste(sex, tissue, sep = "_"),
         lfcpadj = paste(lfc, padj, sep = ", "))  %>%
  dplyr::select(sextissue,comparison, gene, lfcpadj) %>%
  pivot_wider(names_from = comparison, values_from = lfcpadj) 

wideDEGscandidates <- wideDEGs %>%
  filter(gene %in% GOgenesLong$gene)
wideDEGscandidates

wideDEGsBldg <- wideDEGscandidates %>%
  select(sextissue,gene, contains("bldg"))
wideDEGsBldg

wideDEGsControl <- wideDEGscandidates %>%
  select(sextissue,gene, contains("control"))
wideDEGsControl


# sequential in 2 or more comparisons
wideDEGsSequential <- wideDEGscandidates %>%
  select(sextissue,gene, "control_bldg", "bldg_lay" #, 
         #"lay_inc.d3", "inc.d3_inc.d9",
         #"inc.d9_inc.d17", "inc.d17_hatch", "hatch_n5", "n5_n9"
         ) %>%
  mutate(totalNA = rowSums(is.na(.))) %>%
  filter(totalNA < 8) %>%
  arrange(totalNA, gene, sextissue) %>%
  arrange( gene, sextissue) %>%
  select(-totalNA) %>% 
  replace(., is.na(.), "NS")
wideDEGsSequential

```

### manipuluation

```{r}
head(allDEG)

filteredDEGs <- allDEG %>%
  filter(lfc > 0.14 | lfc < -0.14 )
dim(allDEG)
dim(filteredDEGs)

makemanipulationDEGtables <- function(whichcomparisons){
  df <- filteredDEGs %>%
    filter(comparison %in% whichcomparisons) %>%
     mutate(posneg = ifelse(lfc >= 0, "+", "-"),
         sex = recode(sex, "female" = "F", "male" = "M" ),
         tissue = recode(tissue, 
                         "hypothalamus" = "H",
                         "pituitary" = "P", "gonad" = "G"),
         group = paste(sex, tissue, sep = "")) %>%
  mutate(res = paste("(", posneg, ")", sep = "")) %>%
  mutate(compres = paste(comparison, res, sep = "")) %>%
  select(gene, group, compres) %>%
  group_by(group,gene) %>%
  summarize(results = str_c(compres, collapse = "; ")) %>%
  mutate(n = (str_count(results, pattern = ";"))+1) %>%
  arrange(desc(n))  %>%
  filter(n>1) %>%
  group_by(group,results) %>%
  summarize(genes = str_c(gene, collapse = "; "))  %>%
  #arrange(group, results) %>%
  mutate(n = (str_count(genes, pattern = ";"))+1) %>%
  arrange(desc(n))
  print(head(df))
return(df) 
}

removalcomps <- c("inc.d3_m.inc.d3", "inc.d9_m.inc.d9", "inc.d17_m.inc.d17", "hatch_m.n2")
early <- c("early_inc.d9", "hatch_early")
late <- c("hatch_prolong", "inc.d17_prolong")


removalDEGs <- makemanipulationDEGtables(removalcomps)
hatchDEGs <- makemanipulationDEGtables(early)
prolongDEGs <- makemanipulationDEGtables(late)

manipDEGs <- rbind(removalDEGs, hatchDEGs) %>%
  rbind(., prolongDEGs) %>%
  arrange(desc(n))
head(manipDEGs)
```



## save files

```{r write}
#write.csv(allDEG, "../results/03_allDEG.csv", row.names = F)
#write.csv(candidatevsd, "../results/03_candidatevsd.csv")
#write.csv(wideDEGsSequential, "../results/table1.csv", row.names = F)

#write.csv(removalDEGs, "../results/removalDEGs.csv")
#write.csv(hatchDEGs, "../results/hatchDEGs.csv")
#write.csv(prolongDEGs, "../results/prolongDEGs.csv")

#write.csv(manipDEGs, "../results/manipDEGs.csv")

```


### removal overlap

```{r}

filteredDEGs <- allDEG %>%
  filter(lfc > 0.14 | lfc < -0.14 ) %>%
  filter(tissue == "hypothalamus")

removalcomps <- c("inc.d3_m.inc.d3", "inc.d9_m.inc.d9", "inc.d17_m.inc.d17", "hatch_m.n2")

hatch_m.n2DEGs <- filteredDEGs %>%
  filter(comparison == "hatch_m.n2") %>%
  drop_na() %>%
  pull(gene)

inc.d17_m.inc.d17DEGs <- filteredDEGs %>%
  filter(comparison == "inc.d17_m.inc.d17") %>%
  drop_na() %>%
  pull(gene)

inc.d3_m.inc.d3DEGs <- filteredDEGs %>%
  filter(comparison == "inc.d3_m.inc.d3") %>%
  drop_na() %>%
  pull(gene)

inc.d9_m.inc.d9DEGs <- filteredDEGs %>%
  filter(comparison == "inc.d9_m.inc.d9") %>%
  drop_na() %>%
  pull(gene)


intersect(hatch_m.n2DEGs, inc.d17_m.inc.d17DEGs) %>%
  intersect(.,inc.d3_m.inc.d3DEGs) %>%
  intersect(.,inc.d9_m.inc.d9DEGs)

venn.diagram(
  x = list(inc.d3_m.inc.d3DEGs,hatch_m.n2DEGs,
            inc.d9_m.inc.d9DEGs, inc.d17_m.inc.d17DEGs),
  category.names = c("Inc 3" , "Hatch" , "Inc 9", " Inc 17"),
  filename = '../figures/venn-hyp.png',
  output=FALSE,
  print.mode = "raw",
  imagetype = "png",
  col=c("#CDCDCD", '#262625', '#959595','#959595'),
  fill = c("#CDCDCD", "#262625", "#959595", "#959595"),
  main = "Hypothalamus - Offspring Removal",
  sub = "Differentially expressed genes relative to internal controls"
  )


```

## chicks / nesting care


```{r}

filteredDEGs <- allDEG %>%
  filter(lfc > 1.1 | lfc < -1.1 ) %>%
  filter(tissue == "hypothalamus")

chicks <- c("bldg_hatch", "bldg_n5", "bldg_n9",
            "bldg_extend", "bldg_early")

bldg_hatchDEGs <- filteredDEGs %>%
  filter(comparison == "bldg_hatch") %>%
  drop_na() %>%
  pull(gene)

bldg_n5DEGs <- filteredDEGs %>%
  filter(comparison == "bldg_n5") %>%
  drop_na() %>%
  pull(gene)

bldg_n9DEGs <- filteredDEGs %>%
  filter(comparison == "bldg_n9") %>%
  drop_na() %>%
  pull(gene)

bldg_extendDEGs <- filteredDEGs %>%
  filter(comparison == "bldg_extend") %>%
  drop_na() %>%
  pull(gene)

bldg_earlyDEGs <- filteredDEGs %>%
  filter(comparison == "bldg_early") %>%
  drop_na() %>%
  pull(gene)


# chicks
venn.diagram(
  x = list(bldg_n5DEGs, bldg_hatchDEGs, 
           bldg_earlyDEGs,bldg_extendDEGs,  bldg_n9DEGs),
  category.names = c( "N5" ,"Hatch" , "Early", " Extend", "N9"),
  filename = '../figures/venn-chicks-pit.png',
  output=FALSE,
  print.mode = "raw",
  imagetype = "png",
  height = 1500,
	width = 1750,
  resolution = 500,
  cex = 0.75,
  cat.cex = 0.75,
  col=c('#3182bd', "#6baed6",  '#cbc9e2', '#6a51a3', '#08519c'),
  fill = c("#3182bd", "#6baed6", "#cbc9e2", '#6a51a3', "#08519c" )
  )

```


## eggs / incubation

```{r}

filteredDEGs <- allDEG %>%
  filter(lfc > 1.1 | lfc < -1.1 ) %>%
  filter(tissue == "hypothalamus") 

eggs <- c("bldg_lay", 
         "bldg_inc.d3", "bldg_inc.d9", "bldg_inc.d17",
         "bldg_prolong")

bldg_layDEGs <- filteredDEGs %>%
  filter(comparison == "bldg_lay") %>%
  drop_na() %>%
  pull(gene)

bldg_inc.d3DEGs <- filteredDEGs %>%
  filter(comparison == "bldg_inc.d3") %>%
  drop_na() %>%
  pull(gene)

bldg_inc.d9DEGs <- filteredDEGs %>%
  filter(comparison == "bldg_inc.d9") %>%
  drop_na() %>%
  pull(gene)

bldg_inc.d17DEGs <- filteredDEGs %>%
  filter(comparison == "bldg_inc.d17") %>%
  drop_na() %>%
  pull(gene)

bldg_prolongDEGs <- filteredDEGs %>%
  filter(comparison == "bldg_prolong") %>%
  drop_na() %>%
  pull(gene)

# chicks
venn.diagram(
  x = list(bldg_inc.d3DEGs, bldg_layDEGs,  bldg_prolongDEGs,
           bldg_inc.d17DEGs, bldg_inc.d9DEGs),
  category.names = c("Inc3" , "Lay" , "Prolong", " Inc17", "Inc9"),
  filename = '../figures/venn-eggs-pit.png',
  output=FALSE,
  print.mode = "raw",
  imagetype = "png",
  height = 1500,
	width = 1750,
  resolution = 500,
  cex = 0.75,
  cat.cex = 0.75,
  col=c( '#78c679', "#fed98e",  '#9e9ac8','#006837', '#31a354'),
  fill = c("#78c679", "#fed98e",  "#9e9ac8",'#006837', "#31a354")
  )

```


