---
title: "02_DESeq_characterization"
output: md_document
---

```{r setup, message=F, comment=FALSE, warning=FALSE, results='hide'}
library(tidyverse)
library(DESeq2)
library(cowplot)
library(RColorBrewer)
library(pheatmap)
library(kableExtra)
library(viridis)
library(ggimage)
library(knitr)
library(kableExtra)


library(png)
library(grid)
library(ggimage)
library(ggpubr)

library(BiocParallel)
register(MulticoreParam(6))

# load custom functions  
source("../R/themes.R") 
source("../R/functions.R") 

knitr::opts_chunk$set(fig.path = '../figures/characterization/', cache = TRUE)
```

## Characterization data

```{r readfiles}
# import "colData" which contains sample information and "countData" which contains read counts
c.colData <- read.csv("../metadata/00_colData_characterization.csv", header = T, row.names = 1)
c.countData <- read.csv("../results/00_countData_characterization.csv", header = T, row.names = 1)
geneinfo <- read.csv("../metadata/00_geneinfo.csv", row.names = 1)

# set levels
c.colData$treatment <- factor(c.colData$treatment, levels = 
                              c("control",  "bldg", "lay", "inc.d3", "inc.d9", "inc.d17", "hatch", "n5", "n9"))
levels(c.colData$treatment)
c.colData$sextissue <- as.factor(paste(c.colData$sex, c.colData$tissue, sep = "_"))
summary(c.colData[c(7,3,4,5,8)])

geneinfo <- read.csv("../metadata/00_geneinfo.csv", row.names = 1)
head(geneinfo)
```



## Run DESeq on all subsets of the data

```{r dds, cache = T, eval = T}
dds.female_hypothalamus <- subsetDESeq(c.colData,  c.countData, "female_hypothalamus")
dds.female_pituitary <- subsetDESeq(c.colData,  c.countData, "female_pituitary" )
dds.female_gonad <- subsetDESeq(c.colData,  c.countData, "female_gonad" )
dds.male_hypothalamus <- subsetDESeq(c.colData,  c.countData, "male_hypothalamus" )
dds.male_pituitary <- subsetDESeq(c.colData,  c.countData, "male_pituitary"  )
dds.male_gondad <- subsetDESeq(c.colData,  c.countData, "male_gonad")
```



## Plot only a subset of DEGs that relate to hypotheses as bar and line graphs

```{r barplots}
# bars with lines
g <- plotserialDEGs(DEGs.female_hypothalamus, "Female hypothalamus", "#00A08A")
h <- plotserialDEGs(DEGs.female_pituitary, "Female pituitary", "#00A08A")
i <- plotserialDEGs(DEGs.female_gonad, "Female gonad", "#00A08A")
j <- plotserialDEGs(DEGs.male_hypothalamus, "Male hypothalamus", "#F98400")
k <- plotserialDEGs(DEGs.male_pituitary, "Male pituitary", "#F98400")
l <- plotserialDEGs(DEGs.male_gondad, "Male gonad", "#F98400")

mybarplots <- plot_grid(g, h , i , j ,k , l, nrow = 2) 
mybarplots

pdf("../figures/characterization/barplots.pdf", width = 10, height = 6)
plot(mybarplots)
dev.off()
```


```{r linegraphs}

# lines only, different colors for different sex tissue groups
FH <- subsetDEGs(DEGs.female_hypothalamus, "Female hypothalamus")
FP <- subsetDEGs(DEGs.female_pituitary, "Female pituitary")
FG <- subsetDEGs(DEGs.female_gonad, "Female gonad")
MH <- subsetDEGs(DEGs.male_hypothalamus, "Male hypothalamus")
MP <- subsetDEGs(DEGs.male_pituitary, "Male pituitary")
MG <- subsetDEGs(DEGs.male_gondad, "Male gonad")

specificDEGs <- rbind(FH, FP, FG, MH, MP, MG)

specificDEGs$sextissue <- factor(specificDEGs$sextissue, levels = c("Female hypothalamus", "Female pituitary", "Female gonad",
                                                                    "Male hypothalamus", "Male pituitary", "Male gonad"))
mylinegraph <- ggplot(specificDEGs, aes(x=comparison, y=V3, group = sextissue)) +
    geom_line(aes(color=sextissue)) +
    geom_point(aes(color=sextissue)) + 
    theme_minimal(base_size = 10) +
    #theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    labs(x = "Parental stage transitions", y= "No. of differentially expressed genes ",
         title = "Gene expression changes during parental care transitions") + 
    scale_x_discrete(labels=c("Control to\nNest bldg", "Nest bldg. to\nLay", "Lay to\nInc. day 3",
                              "Inc. day 3 to\nInc. day 9", "Inc. day 9 to\nInc. day 17",
                              "Inc. day 17 to\nHatch", "Hatch to\nNestling day 5", "Nestling day 5 to\n Nestling day 9")) +
   theme(legend.title = element_blank(),
         legend.position = c(0.8, 0.8))
mylinegraph

pdf("../figures/characterization/linegraphs.pdf", width = 8, height = 4)
plot(mylinegraph)
dev.off()
```


## Calculate and plot principal component

```{r pca, eval = T, cache = TRUE}

FhypPCA <- returnPCAs(dds.female_hypothalamus)
plotPC12(FhypPCA, "female hypothalamus")

FpitPCA <- returnPCAs(dds.female_pituitary)      
plotPC12(FpitPCA, "female pituitary")

FgonPCA <- returnPCAs(dds.female_gonad)
plotPC12(FgonPCA, "female gonad")

MhypPCA <- returnPCAs(dds.male_hypothalamus)
plotPC12(MhypPCA, "male hypothalamus")

MpitPCA <- returnPCAs(dds.male_pituitary)
plotPC12(MpitPCA, "male pituitary")

MgonPCA <- returnPCAs(dds.male_gondad)
plotPC12(MgonPCA, "male gonad")
```

## heamap of candidate genes

```{r pheatmap, eval = T, cache = TRUE}
colDataFH <- subsetcolData(c.colData, "female_hypothalamus")
colDataFP <- subsetcolData(c.colData, "female_pituitary")
colDataFG <- subsetcolData(c.colData, "female_gonad")
colDataMH <- subsetcolData(c.colData, "male_hypothalamus")
colDataMP <- subsetcolData(c.colData, "male_pituitary")
colDataMG <- subsetcolData(c.colData, "male_gonad")


makepheatmap(dds.female_hypothalamus, colDataFH, "female hypothalamus")
makepheatmap(dds.female_pituitary, colDataFP, "female pituitary")
makepheatmap(dds.female_gonad, colDataFG, "female gonad")
makepheatmap(dds.male_hypothalamus, colDataMH,  "male hypothalamus")
makepheatmap(dds.male_pituitary, colDataMP, "male pituitary")
makepheatmap(dds.male_gondad, colDataMG, "male gonad")
```
## plot candidate genes

```{r candidates, eval = T}
plotcandidates(dds.female_hypothalamus, colDataFH,"female hypothalamus")
plotcandidates(dds.female_pituitary, colDataFP, "female pituitary")
plotcandidates(dds.female_gonad, colDataFG,"female gonad")
plotcandidates(dds.male_hypothalamus, colDataMH,"male hypothalamus")
plotcandidates(dds.male_pituitary,colDataMP, "male pituitary")
plotcandidates(dds.male_gondad,colDataMG, "male gonad")
```

## plot prolactin

```{r prolactin, eval = T}
a <- plotprolactin(dds.female_hypothalamus, colDataFH,"female hypothalamus")
b <- plotprolactin(dds.female_pituitary, colDataFP, "female pituitary")
c <- plotprolactin(dds.female_gonad, colDataFG,"female gonad")
d <- plotprolactin(dds.male_hypothalamus, colDataMH,"male hypothalamus")
e <- plotprolactin(dds.male_pituitary,colDataMP, "male pituitary")
f <- plotprolactin(dds.male_gondad,colDataMG, "male gonad")

mylegend <- get_legend(f)

plots <- plot_grid(a + theme(legend.position = "none"),
          b + theme(legend.position = "none"),
          c + theme(legend.position = "none"),
          d + theme(legend.position = "none"), 
          e + theme(legend.position = "none"),
          f + theme(legend.position = "none"))
plot_grid(plots, mylegend, nrow = 2, rel_heights = c(1,0.1))

```


## correlation heat maps

```{r correlationheatmaps, eval = F}
plotcorrelationheatmaps(dds.female_hypothalamus, colData.female_hypothalamus ,"female hypothalamus: hatch, nest bldg., lay and day-before-hatch cluster")
plotcorrelationheatmaps(dds.female_pituitary, colData.female_pituitary, "female pituitary: clustering suggests an internal clock effect")
plotcorrelationheatmaps(dds.female_gonad, colData.female_gonad,  "female gonad: highly correlated across incubation and early nestling care")
plotcorrelationheatmaps(dds.male_hypothalamus, colData.male_hypothalamus, "male hypothalamus: 100% correlated across most parental care stages")
plotcorrelationheatmaps(dds.male_pituitary, colData.male_pituitary, "male pituitary: non-parental males are most different")
plotcorrelationheatmaps(dds.male_gondad, colData.male_gondad, "male gonad: non-parental males are most different")
```


## DEGs

```{r createDEGs}

createDEGdf <- function(mydds, whichfactor, up, down){
  res <- results(mydds, contrast =c(whichfactor, up, down),
                 independentFiltering = T, alpha = 0.1)
   data <- data.frame(entrezid = row.names(res),
                     padj = res$padj, 
                     logpadj = -log10(res$padj),
                     lfc = res$log2FoldChange)
  data <- na.omit(data)
  
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 0 & data$padj < 0.1, 
                                     yes = up, no = ifelse(data$lfc < 0 & data$padj < 0.1, 
                                                 yes = down, no = "NS")))
  data$direction <- factor(data$direction, levels = c(down, "NS", up))
  
  data <- left_join(data, geneinfo) %>%
    mutate(gene = Name) %>%
    select(gene, lfc, padj, logpadj, direction) %>%
    arrange(desc(lfc))
  
  DEGs <- data %>% filter(direction != "NS")
  print(str(DEGs))
  
  myfilename = paste("../results/03_DEGs", down,up, "csv", sep = ".")
  
  # write dataframe of only significant genes
  write.csv(DEGs, myfilename, row.names = F)
  
  # return data frome with all data, included NS genes
  return(data)
}  

control.bldg <- createDEGdf(dds.female_pituitary, "treatment", "bldg", "control")
bldg.lay <- createDEGdf(dds.female_pituitary, "treatment", "lay", "bldg")
lay.inc.d3 <- createDEGdf(dds.female_pituitary, "treatment", "inc.d3", "lay") 
inc.d3.inc.d9 <- createDEGdf(dds.female_pituitary, "treatment", "inc.d9", "inc.d3") 
inc.d9.inc.d17 <- createDEGdf(dds.female_pituitary, "treatment", "inc.d17", "inc.d9")
inc.d17.hatch <- createDEGdf(dds.female_pituitary, "treatment", "hatch", "inc.d17") 
hatch.n5 <- createDEGdf(dds.female_pituitary, "treatment", "n5", "hatch") 
n5.n9 <- createDEGdf(dds.female_pituitary, "treatment", "n9", "n5") 
```

```{r plotDEGs}
plot.volcano <- function(data, whichfactor, up, down, mycolors){
  
  volcano <- data %>%
    ggplot(aes(x = lfc, y = logpadj)) + 
    geom_point(aes(color = direction), size = 1, alpha = 0.75, na.rm = T) + 
    theme_minimal() +
    scale_color_manual(values = mycolors,
                       name = " ",
                       drop = FALSE) +
    ylim(c(0,30)) +  
    xlim(c(-8,8)) +
    labs(y = "-log10(p)", x = " ")  +
    theme(legend.position = "none",
          legend.direction = "horizontal",
          legend.spacing.x = unit(-0.1, 'cm'),
          legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm"),
          panel.grid = element_blank()) 
  return(volcano)
}

a <- plot.volcano(control.bldg, "treatment", "bldg", "control", colorsvolcano)
b <- plot.volcano(bldg.lay, "treatment", "lay", "bldg", colorsvolcano)
c <- plot.volcano(lay.inc.d3, "treatment", "inc.d3", "lay", colorsvolcano) 
d <- plot.volcano(inc.d3.inc.d9, "treatment", "inc.d9", "inc.d3", colorsvolcano) 
e <- plot.volcano(inc.d9.inc.d17, "treatment", "inc.d17", "inc.d9", colorsvolcano)
f <- plot.volcano(inc.d17.hatch, "treatment", "hatch", "inc.d17", colorsvolcano) 
g <- plot.volcano(hatch.n5, "treatment", "n5", "hatch", colorsvolcano) 
h <- plot.volcano(n5.n9, "treatment", "n9", "n5", colorsvolcano) 


females <- plot_grid(a + 
                       theme(axis.title.x = element_blank(), axis.text.x = element_blank()) +
                       labs(y = "female pituitary \n -log10(p)"),
          b + theme(axis.title = element_blank(), axis.text = element_blank()),
          c + theme(axis.title = element_blank(), axis.text = element_blank()),
          d + theme(axis.title = element_blank(), axis.text = element_blank()), 
          e + theme(axis.title = element_blank(), axis.text = element_blank()),
          f + theme(axis.title = element_blank(), axis.text = element_blank()),
          g + theme(axis.title = element_blank(), axis.text = element_blank()),
          h + theme(axis.title = element_blank(), axis.text = element_blank()), 
          nrow = 1, rel_widths = c(1.75,1,1,1,1,1,1,1,1))
females 
```

## Go terms

```{r GOterms}
reproduction <- read.table("../results/goterms/reproduction.txt", sep = "\t", row.names = NULL,  fill=TRUE)
reproduction$GO <- "reproduction, GO:0000003"  
  
parentalbehavior <- read.table("../results/goterms/parentalbehavior.txt", sep = "\t", row.names = NULL,  fill=TRUE)
parentalbehavior$GO <- "parental behavior, GO:0060746"  

behavior <- read.table("../results/goterms/parentalbehavior.txt", sep = "\t", row.names = NULL,  fill=TRUE)
behavior$GO <- "behavior, GO:0007610" 

systemdevelopment <- read.table("../results/goterms/systemdevelopment.txt", sep = "\t", row.names = NULL,  fill=TRUE)
systemdevelopment$GO <- "system development, GO:0048731" 

endocrineprocess <- read.table("../results/goterms/endocrineprocess.txt", sep = "\t", row.names = NULL,  fill=TRUE)
endocrineprocess$GO <- "endocrine process, GO:0050886" 

responsetostimulus <- read.table("../results/goterms/responsetostimulus.txt", sep = "\t", row.names = NULL,  fill=TRUE)
responsetostimulus$GO <- "response to external stimulus, GO:0009605" 


GOterms <- rbind(parentalbehavior, reproduction, behavior, 
                 systemdevelopment, endocrineprocess, responsetostimulus)

GOterms <- GOterms %>%
  dplyr::mutate(gene = toupper(MGI.Gene.Marker.ID)) %>% 
  dplyr::select(gene, GO) %>% 
  dplyr::distinct(gene, GO) %>% 
 group_by(gene) 

head(GOterms)

returnGOgenes <- function(whichGOterms, data, mycomparison){
  
  df <- inner_join(whichGOterms, data) %>% 
    filter(direction != "NS") %>%  
    select(gene, GO) %>% 
    group_by(GO) %>% 
    summarize(genes = str_c(gene, collapse = ", "))  %>% 
    mutate(comparison = mycomparison) %>% 
    select(comparison, GO, genes)
  
  return(df)
}  
  
dt1 <- returnGOgenes(GOterms,control.bldg, "control v bldg") 
dt2 <- returnGOgenes(GOterms,bldg.lay , "bldg v lay") 
dt3 <- returnGOgenes(GOterms,lay.inc.d3 , "lay v inc d3")  
dt4 <- returnGOgenes(GOterms,inc.d3.inc.d9 , "inc d3 v inc d9")  
dt5 <- returnGOgenes(GOterms,inc.d9.inc.d17 , "inc d9 v inc d17")  
kable(dt5) 

dt6 <- returnGOgenes(GOterms,inc.d17.hatch, "inc d17 v hatch")  
dt7 <- returnGOgenes(GOterms,hatch.n5, "hatch v n5")  
kable(dt7) 

dt8 <- returnGOgenes(GOterms,n5.n9, "n5 v n9") 

alldt <- rbind(dt1,dt2,dt3,dt4,dt5,dt6,dt7,dt8)
alldt

write.csv(alldt, "../results/03_GO_femalepituitary.csv")
```

## males and females

```{r createDEGsmales}
control.bldg2 <- createDEGdf(dds.male_pituitary, "treatment", "bldg", "control")
bldg.lay2 <- createDEGdf(dds.male_pituitary, "treatment", "lay", "bldg")
lay.inc.d32 <- createDEGdf(dds.male_pituitary, "treatment", "inc.d3", "lay") 
inc.d3.inc.d92 <- createDEGdf(dds.male_pituitary, "treatment", "inc.d9", "inc.d3") 
inc.d9.inc.d172 <- createDEGdf(dds.male_pituitary, "treatment", "inc.d17", "inc.d9")
inc.d17.hatch2 <- createDEGdf(dds.male_pituitary, "treatment", "hatch", "inc.d17") 
hatch.n52 <- createDEGdf(dds.male_pituitary, "treatment", "n5", "hatch") 
n5.n92 <- createDEGdf(dds.male_pituitary, "treatment", "n9", "n5") 

```



```{r plotDEGsmales}
a2 <- plot.volcano(control.bldg2, "treatment", "bldg", "control", colorsvolcano)
b2 <- plot.volcano(bldg.lay2, "treatment", "lay", "bldg", colorsvolcano)
c2 <- plot.volcano(lay.inc.d32, "treatment", "inc.d3", "lay", colorsvolcano) 
d2 <- plot.volcano(inc.d3.inc.d92, "treatment", "inc.d9", "inc.d3", colorsvolcano) 
e2 <- plot.volcano(inc.d9.inc.d172, "treatment", "inc.d17", "inc.d9", colorsvolcano)
f2 <- plot.volcano(inc.d17.hatch2, "treatment", "hatch", "inc.d17", colorsvolcano) 
g2 <- plot.volcano(hatch.n52, "treatment", "n5", "hatch", colorsvolcano) 
h2 <- plot.volcano(n5.n92, "treatment", "n9", "n5", colorsvolcano) 

males <- plot_grid(a2 + labs(y = "male pituitary \n -log10(p)") + theme(axis.text.x = element_blank()),
          b2 + theme(axis.title.y = element_blank(), axis.text = element_blank()),
          c2 + theme(axis.title.y = element_blank(), axis.text = element_blank()),
          d2 + theme(axis.title.y = element_blank(), axis.text = element_blank()), 
          e2 + theme(axis.title.y = element_blank(), axis.text = element_blank()), 
          f2 + theme(axis.title.y = element_blank(), axis.text = element_blank()),
          g2 + theme(axis.title.y = element_blank(), axis.text = element_blank()),
          h2 + theme(axis.title.y = element_blank(), axis.text = element_blank()), 
          nrow = 1, rel_widths = c(1.75,1,1,1,1,1,1,1,1))
males
```

### tissue levels, ignoring sex differences, sort of

```{r gonaddfs, eval=T}

dds.gonads <- subsetDESeq2(c.colData,  c.countData, c("female_gonad","male_gonad"))
```

```{r gonaddfs2, eval=T}
control.bldg <- createDEGdf(dds.gonads, "treatment", "bldg", "control")
bldg.lay <- createDEGdf(dds.gonads, "treatment", "lay", "bldg")
lay.inc.d3 <- createDEGdf(dds.gonads, "treatment", "inc.d3", "lay") 
inc.d3.inc.d9 <- createDEGdf(dds.gonads, "treatment", "inc.d9", "inc.d3") 
inc.d9.inc.d17 <- createDEGdf(dds.gonads, "treatment", "inc.d17", "inc.d9")
inc.d17.hatch <- createDEGdf(dds.gonads, "treatment", "hatch", "inc.d17") 
hatch.n5 <- createDEGdf(dds.gonads, "treatment", "n5", "hatch") 
n5.n9 <- createDEGdf(dds.gonads, "treatment", "n9", "n5") 
```

```{r gonplot, eval=T}
a2 <- plot.volcano(control.bldg, "treatment", "bldg", "control", colorsvolcano)
b2 <- plot.volcano(bldg.lay, "treatment", "lay", "bldg", colorsvolcano)
c2 <- plot.volcano(lay.inc.d3, "treatment", "inc.d3", "lay", colorsvolcano) 
d2 <- plot.volcano(inc.d3.inc.d9, "treatment", "inc.d9", "inc.d3", colorsvolcano) 
e2 <- plot.volcano(inc.d9.inc.d17, "treatment", "inc.d17", "inc.d9", colorsvolcano)
f2 <- plot.volcano(inc.d17.hatch, "treatment", "hatch", "inc.d17", colorsvolcano) 
g2 <- plot.volcano(hatch.n5, "treatment", "n5", "hatch", colorsvolcano) 
h2 <- plot.volcano(n5.n9, "treatment", "n9", "n5", colorsvolcano) 


gonads <- plot_grid(a2 + labs(y = "gonads \n -log10(p)"),
          b2 + theme(axis.title.y = element_blank(), axis.text.y = element_blank()),
          c2 + theme(axis.title.y = element_blank(), axis.text.y = element_blank()),
          d2 + theme(axis.title.y = element_blank(), axis.text.y = element_blank()) +
            labs(x = "log fold change"), 
          e2 + theme(axis.title.y = element_blank(), axis.text.y = element_blank()), 
          f2 + theme(axis.title.y = element_blank(), axis.text.y = element_blank()),
          g2 + theme(axis.title.y = element_blank(), axis.text.y = element_blank()),
          h2 + theme(axis.title.y = element_blank(), axis.text.y = element_blank()), 
          nrow = 1, rel_widths = c(1.75,1,1,1,1,1,1,1,1))
gonads
```

```{r hypdfs, eval=T}
dds.hypothalamus <- subsetDESeq2(c.colData,  c.countData, c("female_hypothalamus","male_hypothalamus"))
```

```{r hypdfs2, eval=T}
control.bldg <- createDEGdf(dds.hypothalamus, "treatment", "bldg", "control")
bldg.lay <- createDEGdf(dds.hypothalamus, "treatment", "lay", "bldg")
lay.inc.d3 <- createDEGdf(dds.hypothalamus, "treatment", "inc.d3", "lay") 
inc.d3.inc.d9 <- createDEGdf(dds.hypothalamus,  "treatment", "inc.d9", "inc.d3") 
inc.d9.inc.d17 <- createDEGdf(dds.hypothalamus, "treatment", "inc.d17", "inc.d9")
inc.d17.hatch <- createDEGdf(dds.hypothalamus, "treatment", "hatch", "inc.d17") 
hatch.n5 <- createDEGdf(dds.hypothalamus, "treatment", "n5", "hatch") 
n5.n9 <- createDEGdf(dds.hypothalamus, "treatment", "n9", "n5") 
```

```{r hypplot, eval = T}
a <- plot.volcano(control.bldg, "treatment", "bldg", "control", colorsvolcano)
b <- plot.volcano(bldg.lay, "treatment", "lay", "bldg", colorsvolcano)
c <- plot.volcano(lay.inc.d3, "treatment", "inc.d3", "lay", colorsvolcano) 
d <- plot.volcano(inc.d3.inc.d9, "treatment", "inc.d9", "inc.d3", colorsvolcano) 
e <- plot.volcano(inc.d9.inc.d17, "treatment", "inc.d17", "inc.d9", colorsvolcano)
f <- plot.volcano(inc.d17.hatch, "treatment", "hatch", "inc.d17", colorsvolcano) 
g <- plot.volcano(hatch.n5, "treatment", "n5", "hatch", colorsvolcano) 
h <- plot.volcano(n5.n9, "treatment", "n9", "n5", colorsvolcano) 


hypothalamus <- plot_grid(a + 
                       theme(axis.title.x = element_blank(), axis.text.x = element_blank()) +
                       labs(y = "hypothalamus \n -log10(p)"),
          b + theme(axis.title = element_blank(), axis.text = element_blank()),
          c + theme(axis.title = element_blank(), axis.text = element_blank()),
          d + theme(axis.title = element_blank(), axis.text = element_blank()), 
          e + theme(axis.title = element_blank(), axis.text = element_blank()),
          f + theme(axis.title = element_blank(), axis.text = element_blank()),
          g + theme(axis.title = element_blank(), axis.text = element_blank()),
          h + theme(axis.title = element_blank(), axis.text = element_blank()), 
          nrow = 1,  rel_widths = c(1.75,1,1,1,1,1,1,1,1))
hypothalamus

```

```{r allvolcanos, fig.height=8}
background <- png::readPNG("../figures/images/DoveParentsRNAseq_charicons.png")
icons <- ggdraw() +  draw_image(background, scale = 1)

plot_grid(icons, hypothalamus, females, males, gonads, nrow = 5, rel_heights = c(1,1,1,1,1.25))

```

