---
title: "07_topconfects"
output: md_document
---

## Based on a discussion on Twitter, https://twitter.com/Jared_Mamrot/status/1161047070821646336, I decided to try out the R package `topconfects` to test the effect size. 

```{r setup, message=F, comment=FALSE, warning=FALSE, results='hide'}
library(DESeq2)
library(tidyverse)
library(topconfects)

knitr::opts_chunk$set(fig.path = '../figures/topconfects/', cache = TRUE, message = F, echo = T)
```

## This is a demo from the vignette showing how the tool works. 

```{r vignette}
# Confident log2 fold changes based on a DESeq2 analysis

# Generate some random data
n <- 20
folds <- seq(-8,8,length.out=n)
row_means <- runif(n, min=0, max=5)
lib_scale <- c(1,2,3,4)
means <- 2^(outer(folds, c(-0.5,-0.5,0.5,0.5))) *
  row_means * rep(lib_scale,each=n)
counts <- rnbinom(length(means), mu=means, size=1/0.1)
dim(counts) <- dim(means)
group <- factor(c("A","A","B","B"))
# Apply DESeq2
dds <- DESeqDataSetFromMatrix(
  countData = counts,
  colData = data.frame(group=group),
  design = ~group)
dds <- DESeq(dds)
# Find top confident effect sizes
deseq2_confects(dds, name="group_B_vs_A", step=0.1)

#deseq2_confects(dds, name="group_A_vs_B", step=0.1) # Error: subscript contains invalid names

```

## Now I import the gene expression data and order the treatment according the the parentatl care cycle

```{r readfiles}
# import "colData" which contains sample information and "countData" which contains read counts
c.colData <- read.csv("../metadata/00_colData_characterization.csv", header = T, row.names = 1)
c.countData <- read.csv("../results/00_countData_characterization.csv", header = T, row.names = 1)
geneinfo <- read.csv("../metadata/00_geneinfo.csv", row.names = 1)

# create grouping
c.colData$sextissue <- as.factor(paste(c.colData$sex, c.colData$tissue, sep = "_"))

c.colData$treatment <- factor(c.colData$treatment, levels = 
                              c("control",  "bldg", "lay", "inc.d3", "inc.d9", "inc.d17", "hatch", "n5", "n9"))
levels(c.colData$treatment)
```

## This is a function to subset the data to look compare one transition (eg. incubation day 17 to lay) in a single "sex tissue" group (e.g. female pitutiary)

```{r subsetnconfects2}

subsetnconfects <- function(mygroup, mytreatments, mycomparison){
  # subset col and count data
  colData <- c.colData %>%
      dplyr::filter(sextissue == mygroup) %>%
      dplyr::filter(treatment %in% mytreatments) %>%
      droplevels()
  row.names(colData) <- colData$V1
  
  # which counts to save
  savecols <- as.character(colData$V1) 
  savecols <- as.vector(savecols) 
  
  # save counts that match colData
  countData <- c.countData %>% dplyr::select(one_of(savecols)) 
  
  # check that row and col lenghts are equal
  #print(ncol(countData) == nrow(colData))

  # run deseq
  dds <- DESeqDataSetFromMatrix(
    countData = countData,
   colData = colData,
   design = ~treatment)
  dds <- DESeq(dds)
  res <- results(dds)
  print(summary(res))
  # Find top confident effect sizes
  deseq2_confects(dds, name=mycomparison, step=0.1)
}
```


## Number of differentially expressed genes for the building versus lay comparison

```{r treatment_lay_vs_bldg}
subsetnconfects("female_hypothalamus", c("bldg", "lay"), "treatment_lay_vs_bldg") #1
subsetnconfects("male_hypothalamus", c("bldg", "lay"), "treatment_lay_vs_bldg") #0

subsetnconfects("female_pituitary", c("bldg", "lay"), "treatment_lay_vs_bldg") #455
subsetnconfects("male_pituitary", c("bldg", "lay"), "treatment_lay_vs_bldg") #0

subsetnconfects("female_gonad", c("bldg", "lay"), "treatment_lay_vs_bldg") #452
subsetnconfects("male_gonad", c("bldg", "lay"), "treatment_lay_vs_bldg") #26

```

## Number of differentially expressed genes for the incubation day 17 versus hatch comparison

```{r treatment_hatch_vs_inc.d17}
subsetnconfects("female_hypothalamus", c("inc.d17", "hatch"), "treatment_hatch_vs_inc.d17") #0
subsetnconfects("male_hypothalamus", c("inc.d17", "hatch"), "treatment_hatch_vs_inc.d17") #3

subsetnconfects("female_pituitary", c("inc.d17", "hatch"), "treatment_hatch_vs_inc.d17") #6
subsetnconfects("male_pituitary", c("inc.d17", "hatch"), "treatment_hatch_vs_inc.d17") #24

subsetnconfects("female_gonad", c("inc.d17", "hatch"), "treatment_hatch_vs_inc.d17") #3
subsetnconfects("male_gonad", c("inc.d17", "hatch"), "treatment_hatch_vs_inc.d17") #1

```


