---
title: "Fig2"
output: md_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(cowplot)
library(kableExtra) 
library(ggtext) # for italic fonts
library(ggsignif) # fot stats on plots

source("../R/themes.R")

knitr::opts_chunk$set(echo = TRUE, message = F, fig.path = "../figures/")
```

## DEGs

```{r allDEGs}
DEG_path <- "../results/DEseq2/"   # path to the data
DEG_files <- dir(DEG_path, pattern = "*DEGs") # get file names
DEG_pathfiles <- paste0(DEG_path, DEG_files)
#DEG_files

allDEG <- DEG_pathfiles %>%
  setNames(nm = .) %>% 
  map_df(~read_csv(.x), .id = "file_name") %>% 
  mutate(DEG = sapply(strsplit(as.character(file_name),'./results/DEseq2/'), "[", 2))  %>% 
  mutate(DEG = sapply(strsplit(as.character(DEG),'_diffexp.csv'), "[", 1))  %>% 
  mutate(tissue = sapply(strsplit(as.character(DEG),'\\.'), "[", 1)) %>%
  mutate(down = sapply(strsplit(as.character(DEG),'\\_'), "[", 3)) %>%
  mutate(up = sapply(strsplit(as.character(DEG),'\\_'), "[", 4)) %>%
  mutate(comparison = paste(down,up, sep = "_")) %>%
  mutate(sex = sapply(strsplit(as.character(sextissue),'\\_'), "[", 1)) %>%
  mutate(tissue = sapply(strsplit(as.character(sextissue),'\\_'), "[", 2)) %>%
dplyr::select(sex,tissue,comparison, direction, gene, lfc, padj, logpadj) 
head(allDEG)

allDEG$tissue <- factor(allDEG$tissue , levels = tissuelevel)
allDEG$comparison <- factor(allDEG$comparison , levels = comparisonlevels)
allDEG$direction <- factor(allDEG$direction, levels = charlevels)
```



```{r figurefunctions}

makebargraph <- function(whichtissue, myylab, lowlim, higherlim){
  p <- allDEG %>%
    filter(tissue == whichtissue,
           comparison != "control_bldg") %>%
  ggplot(aes(x = comparison,  fill = direction)) +
    geom_bar(position = "dodge") +
    facet_grid(tissue~sex) +
    theme_B3() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          legend.position = "none")  +
    guides(fill = guide_legend(nrow = 1)) +
    labs(x = "Sequential parental care stage comparisons", 
         y = myylab,
         subtitle = " ") +
  scale_fill_manual(values = allcolors,
                       name = " ",
                       drop = FALSE) +
  scale_color_manual(values = allcolors) +
  geom_text(stat='count', aes(label=..count..), vjust =-0.5, 
            position = position_dodge(width = 1),
            size = 2, color = "black")  +
  ylim(lowlim, higherlim)
  return(p)
}

```


## Internal versus external 


```{r hypothesis}
DEG_path <- "../results/DEseq2/hypothesis/"   # path to the data
DEG_files <- dir(DEG_path, pattern = "*DEGs") # get file names
DEG_pathfiles <- paste0(DEG_path, DEG_files)
#DEG_files

allDEG2 <- DEG_pathfiles %>%
  setNames(nm = .) %>% 
  map_df(~read_csv(.x), .id = "file_name") %>% 
  mutate(DEG = sapply(strsplit(as.character(file_name),'./results/DEseq2/hypothesis/'), "[", 2))  %>% 
  mutate(DEG = sapply(strsplit(as.character(DEG),'_diffexp.csv'), "[", 1))  %>% 
  mutate(tissue = sapply(strsplit(as.character(DEG),'\\.'), "[", 1)) %>%
  mutate(down = sapply(strsplit(as.character(DEG),'\\_'), "[", 3)) %>%
  mutate(up = sapply(strsplit(as.character(DEG),'\\_'), "[", 4)) %>%
  mutate(comparison = paste(down,up, sep = "_")) %>%
  mutate(sex = sapply(strsplit(as.character(sextissue),'\\_'), "[", 1)) %>%
  mutate(tissue = sapply(strsplit(as.character(sextissue),'\\_'), "[", 2)) %>%
dplyr::select(sex,tissue,comparison, direction, gene, lfc, padj, logpadj) 
head(allDEG)

allDEG2$tissue <- factor(allDEG2$tissue, levels = tissuelevel)

allDEG2$comparison <- factor(allDEG2$comparison, levels = c("eggs_chicks", "lo_hi"))
allDEG2 <- allDEG2 %>% mutate(comparison = fct_recode(comparison, "lo vs. hi PRL   " = "lo_hi",
                                                      "eggs vs. chicks" = "eggs_chicks"))
allDEG2$direction <- factor(allDEG2$direction, levels = c("eggs", "chicks", "lo", "hi"))



```

## DEGS

```{r fig2, fig.width=6, fig.height=5}
# hyp
p1 <- makebargraph("hypothalamus","DEGs", 0, 1250) + theme(axis.text.x = element_blank(), 
                                                               axis.title.x = element_blank())

# pit
p2 <- makebargraph("pituitary","DEGs", 0, 1250)  +  theme(axis.text.x = element_blank(), axis.title.x = element_blank(), 
                                                              strip.text.x = element_blank())

# gon
p3 <- makebargraph("gonad","DEGs", 0, 1250) +  theme(strip.text.x = element_blank())

bcd <- plot_grid(p1,p2,p3, nrow = 3, rel_heights = c(1.2,1,1.5), labels = c("b", "c", "d"), label_size = 8)


expdesign <- png::readPNG("../figures/images/DoveParentsRNAseq_DEGS.png")
expdesign <- ggdraw() +  draw_image(expdesign, scale = 1)

plot_grid(expdesign, bcd, labels = c("a", " "), label_size = 8, nrow = 2, rel_heights = c(0.2,1))


```


```{r fig2b, fig.width=5, fig.height=5}

makenewbargraph <- function(whichtissue, whichsex,  whichcomparison, lowlim, higherlim){
  p <- allDEG2 %>%
    filter(tissue == whichtissue,
           comparison == whichcomparison,
           sex == whichsex) %>%
    ggplot(aes(x = comparison,  fill = direction)) +
    geom_bar(position = "dodge", drop = FALSE) +
    theme_B3() +
    theme(legend.position = "none")  +
    guides(fill = guide_legend(nrow = 1)) +
    labs( y = whichtissue) +
  geom_text(stat='count', aes(label=..count..), vjust =-0.5, 
           position = position_dodge(width = 1),
           size = 2, color = "black")  + 
      ylim(lowlim, higherlim) +
  scale_fill_manual(values = allcolors, name = "higher in")   + 
    theme(axis.text.x = element_blank()) 
  return(p)
}


b11 <- makenewbargraph("hypothalamus", "female","eggs vs. chicks", 0, 800) +  labs(subtitle = "females", x = NULL, y = "Hyp. DEGS")
b21 <- makenewbargraph("pituitary", "female","eggs vs. chicks", 0, 800)  + labs(x = NULL, y = "Pit. DEGS")
b31 <- makenewbargraph("gonad", "female", "eggs vs. chicks", 0, 800) + labs(x = "eggs vs. chicks", y = "Gon. DEGS")   
b112131 <- plot_grid(b11,b21,b31, nrow = 3, rel_heights = c(1.1,1,1.1))


b12 <- makenewbargraph("hypothalamus", "male",  "eggs vs. chicks", 0, 800) + labs(subtitle = "males", x = NULL)+ 
  theme(axis.title.y = element_blank(), axis.text.y = element_blank())
b22 <- makenewbargraph("pituitary", "male", "eggs vs. chicks", 0, 800) + labs(x = NULL)+ 
  theme(axis.title.y = element_blank(), axis.text.y = element_blank())
b32 <- makenewbargraph("gonad", "male", "eggs vs. chicks", 0, 800) + labs(x = "eggs vs. chicks")   + 
  theme(axis.title.y = element_blank(), axis.text.y = element_blank())
b122232 <- plot_grid(b12, b22,b32, nrow = 3, rel_heights = c(1.1,1,1.1)) 


c11 <- makenewbargraph("hypothalamus", "female", "lo vs. hi PRL   ", 0, 2500) +  labs(subtitle = "females", x = NULL) + 
  theme(axis.title.y = element_blank())
c21 <- makenewbargraph("pituitary", "female", "lo vs. hi PRL   ", 0, 2500)  + labs(x = NULL)+ 
  theme(axis.title.y = element_blank())
c31 <- makenewbargraph("gonad", "female", "lo vs. hi PRL   ", 0, 2500) + labs(x = "lo vs. hi PRL")  + 
  theme(axis.title.y = element_blank()) 
c112131 <- plot_grid(c11,c21,c31, nrow = 3, rel_heights = c(1.1,1,1.1))


c12 <- makenewbargraph("hypothalamus", "male",  "lo vs. hi PRL   ", 0, 2500) + labs(subtitle = "males", x = NULL)+ 
  theme(axis.title.y = element_blank(), axis.text.y = element_blank())
c22 <- makenewbargraph("pituitary", "male", "lo vs. hi PRL   ", 0, 2500) + labs(x = NULL)+ 
  theme(axis.title.y = element_blank(), axis.text.y = element_blank())
c32 <- makenewbargraph("gonad", "male", "lo vs. hi PRL   ", 0, 2500) + labs(x = "lo vs. hi PRL")   + 
  theme(axis.title.y = element_blank(), axis.text.y = element_blank())
c122232 <- plot_grid(c12, c22,c32, nrow = 3, rel_heights = c(1.1,1,1.1)) 


hypothesisbars <-  plot_grid(b112131, b122232, c112131, c122232, nrow = 1, rel_widths = c(1.3, 1, 1.1, 1),
          labels = c("b", " ", "c", " "), label_size = 8)

expdesign2 <- png::readPNG("../figures/images/DoveParentsRNAseq_hypothesis.png")
expdesign2 <- ggdraw() +  draw_image(expdesign2, scale = 1)


plot_grid(expdesign2, hypothesisbars, rel_heights = c(0.5,1), nrow = 2, labels = c("a", " "), label_size = 8)
```

```{r GO}
## shinygo termshttp://bioinformatics.sdstate.edu/go/
allDEG2 %>% filter(tissue == "pituitary", direction == "chicks") %>% arrange(gene) %>% pull(gene) # nervous systme development, neurogenesis, neuron differentiaion 
allDEG2 %>% filter(tissue == "pituitary", direction == "eggs") %>% arrange(gene) %>% pull(gene) # miotic cell cycle, cell cycle, cell cycle process
allDEG2 %>% filter(tissue == "pituitary", direction == "hi") %>% arrange(gene) %>% pull(gene) # this 1000 combined with next 10000
allDEG2 %>% filter(tissue == "pituitary", direction == "hi") %>% arrange(desc(gene)) %>% pull(gene) # Cellular localization, #Cellular protein localization, Protein targeting to ER

#miotic cell cycle, miotic cell cycle process, cell cycle
allDEG2 %>% filter(tissue == "pituitary", direction == "lo") %>% arrange(gene)  %>% pull(gene)   # this 1000 combined with next 10000
allDEG2 %>% filter(tissue == "pituitary", direction == "lo") %>% arrange(desc(gene)) %>% pull(gene) #Anatomical structure morphogenesis, Regulation of developmental process, Regulation of signaling


allDEG2 %>%
  filter(tissue == "pituitary") %>%
  select(sex, tissue, comparison, gene) %>%
  group_by(sex, tissue, gene) %>%
  summarize(groups = str_c(comparison , collapse = " "))   %>%
   group_by(sex, tissue, groups) %>%
  summarize(n = n()) %>%
  arrange(desc(n))

allDEG2 %>%
  filter(tissue == "pituitary") %>%
  select(sex, tissue, comparison, gene) %>%
  group_by(sex, tissue, gene) %>%
  summarize(groups = str_c(comparison , collapse = " "))  %>%
  filter(groups == "eggs vs. chicks") %>%
  arrange(gene) %>%
  distinct(gene) %>%
  pull(gene)  # Nervous system development, Cell development, Anatomical structure morphogenesis

allDEG2 %>%
  filter(comparison == "eggs vs. chicks") %>%
  group_by(sex, tissue, comparison, direction) %>%
  summarize(n = n()) %>%
  arrange(comparison, direction, tissue, sex)

allDEG2 %>%
  filter(comparison == "lo vs. hi PRL   ") %>%
  group_by(sex, tissue, comparison, direction) %>%
  summarize(n = n()) %>%
  arrange(comparison, direction, tissue, sex)

```
```{r}
allDEG2 %>%
  filter(tissue == "pituitary") %>%
  select(sex, tissue, comparison, gene) %>%
  group_by(sex, tissue, gene) %>%
  summarize(groups = str_c(comparison , collapse = " "))  %>%
  filter(groups == "lo vs. hi PRL   ") %>%
  arrange(gene) %>%
  distinct(gene) %>%
  pull(gene)  # Nervous system development, Cell development, Anatomical structure morphogenesis

allDEG2 %>%
  group_by(sex, tissue, comparison, direction) %>%
  summarize(n = n())
```

## total degs

```{r}
allDEG %>%
  group_by(sex, tissue, comparison) %>%
  summarize(totalDEGs = n()) %>%
  arrange(tissue, comparison)
```


## candidate genes 
```{r writefiles}
candidategenes <- c("OXT", "AVP", "GNRH1",  "AR", "POMC", "AGRP",
                       "CRH", "AVPR1A", "AVPR1B", "AVPR2",
                       "CYP19A1", "DRD1", "DRD2", "PRL", "PRLR", "SOX9") 

table1 <- allDEG %>%
  filter(gene %in% candidategenes) %>%
  group_by(sex, tissue, comparison) %>%
  summarize(genes = str_c(gene, collapse = " ")) %>%
  pivot_wider(names_from = comparison, values_from = genes ) %>%
  select(sex, tissue, control_bldg, lay_inc.d3, inc.d3_inc.d9,
        inc.d9_inc.d17, hatch_n5)  %>%
  arrange( tissue, sex)
table1

write_csv(table1, "../results/table1.csv")

suppletable1 <- allDEG %>%
  filter(comparison != "control_bldg") %>%
  group_by(sex, tissue, comparison) %>%
  arrange( tissue, sex)

suppletable1 %>%
  group_by(tissue) %>%
  summarize(totalDEGs = n())

write_csv(table1, "../results/suppletable1.csv")

```